// @aws-sdk/client-sagemaker-runtime/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var eventstreamSerdeConfigResolver = require('@smithy/eventstream-serde-config-resolver');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "sagemaker",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class SageMakerRuntimeClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = eventstreamSerdeConfigResolver.resolveEventStreamSerdeConfig(_config_6);
        const _config_8 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_7);
        const _config_9 = resolveRuntimeExtensions(_config_8, configuration?.extensions || []);
        this.config = _config_9;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultSageMakerRuntimeHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class SageMakerRuntimeServiceException extends smithyClient.ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, SageMakerRuntimeServiceException.prototype);
    }
}

class InternalDependencyException extends SageMakerRuntimeServiceException {
    name = "InternalDependencyException";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "InternalDependencyException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalDependencyException.prototype);
        this.Message = opts.Message;
    }
}
class InternalFailure extends SageMakerRuntimeServiceException {
    name = "InternalFailure";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "InternalFailure",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalFailure.prototype);
        this.Message = opts.Message;
    }
}
class ModelError extends SageMakerRuntimeServiceException {
    name = "ModelError";
    $fault = "client";
    Message;
    OriginalStatusCode;
    OriginalMessage;
    LogStreamArn;
    constructor(opts) {
        super({
            name: "ModelError",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ModelError.prototype);
        this.Message = opts.Message;
        this.OriginalStatusCode = opts.OriginalStatusCode;
        this.OriginalMessage = opts.OriginalMessage;
        this.LogStreamArn = opts.LogStreamArn;
    }
}
class ModelNotReadyException extends SageMakerRuntimeServiceException {
    name = "ModelNotReadyException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ModelNotReadyException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ModelNotReadyException.prototype);
        this.Message = opts.Message;
    }
}
class ServiceUnavailable extends SageMakerRuntimeServiceException {
    name = "ServiceUnavailable";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "ServiceUnavailable",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceUnavailable.prototype);
        this.Message = opts.Message;
    }
}
class ValidationError extends SageMakerRuntimeServiceException {
    name = "ValidationError";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ValidationError",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ValidationError.prototype);
        this.Message = opts.Message;
    }
}
class InternalStreamFailure extends SageMakerRuntimeServiceException {
    name = "InternalStreamFailure";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "InternalStreamFailure",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalStreamFailure.prototype);
        this.Message = opts.Message;
    }
}
class ModelStreamError extends SageMakerRuntimeServiceException {
    name = "ModelStreamError";
    $fault = "client";
    Message;
    ErrorCode;
    constructor(opts) {
        super({
            name: "ModelStreamError",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ModelStreamError.prototype);
        this.Message = opts.Message;
        this.ErrorCode = opts.ErrorCode;
    }
}

const _A = "Accept";
const _B = "Body";
const _BB = "BodyBlob";
const _By = "Bytes";
const _CA = "CustomAttributes";
const _CAH = "CustomAttributesHeader";
const _CSI = "ClosedSessionId";
const _CT = "ContentType";
const _CT_ = "Content-Type";
const _EC = "ErrorCode";
const _EE = "EnableExplanations";
const _EN = "EndpointName";
const _FL = "FailureLocation";
const _ICN = "InferenceComponentName";
const _IDE = "InternalDependencyException";
const _IE = "InvokeEndpoint";
const _IEA = "InvokeEndpointAsync";
const _IEAI = "InvokeEndpointAsyncInput";
const _IEAO = "InvokeEndpointAsyncOutput";
const _IEI = "InvokeEndpointInput";
const _IEO = "InvokeEndpointOutput";
const _IEWRS = "InvokeEndpointWithResponseStream";
const _IEWRSI = "InvokeEndpointWithResponseStreamInput";
const _IEWRSO = "InvokeEndpointWithResponseStreamOutput";
const _IF = "InternalFailure";
const _II = "InferenceId";
const _IL = "InputLocation";
const _IPV = "InvokedProductionVariant";
const _ISF = "InternalStreamFailure";
const _ITS = "InvocationTimeoutSeconds";
const _LSA = "LogStreamArn";
const _M = "Message";
const _ME = "ModelError";
const _MNRE = "ModelNotReadyException";
const _MSE = "ModelStreamError";
const _NSI = "NewSessionId";
const _OL = "OutputLocation";
const _OM = "OriginalMessage";
const _OSC = "OriginalStatusCode";
const _PB = "PartBlob";
const _PP = "PayloadPart";
const _RS = "ResponseStream";
const _RTTLS = "RequestTTLSeconds";
const _SI = "SessionId";
const _SU = "ServiceUnavailable";
const _TCH = "TargetContainerHostname";
const _TM = "TargetModel";
const _TV = "TargetVariant";
const _VE = "ValidationError";
const _XASA = "X-Amzn-SageMaker-Accept";
const _XASCA = "X-Amzn-SageMaker-Custom-Attributes";
const _XASCSI = "X-Amzn-SageMaker-Closed-Session-Id";
const _XASCT = "X-Amzn-SageMaker-Content-Type";
const _XASEE = "X-Amzn-SageMaker-Enable-Explanations";
const _XASF = "X-Amzn-SageMaker-FailureLocation";
const _XASI = "X-Amzn-SageMaker-InputLocation";
const _XASIC = "X-Amzn-SageMaker-Inference-Component";
const _XASII = "X-Amzn-SageMaker-Inference-Id";
const _XASI_ = "X-Amzn-SageMaker-InvocationTimeoutSeconds";
const _XASNSI = "X-Amzn-SageMaker-New-Session-Id";
const _XASO = "X-Amzn-SageMaker-OutputLocation";
const _XASR = "X-Amzn-SageMaker-RequestTTLSeconds";
const _XASSI = "X-Amzn-SageMaker-Session-Id";
const _XASTCH = "X-Amzn-SageMaker-Target-Container-Hostname";
const _XASTM = "X-Amzn-SageMaker-Target-Model";
const _XASTV = "X-Amzn-SageMaker-Target-Variant";
const _aQE = "awsQueryError";
const _c = "client";
const _e = "error";
const _eP = "eventPayload";
const _h = "http";
const _hE = "httpError";
const _hH = "httpHeader";
const _s = "server";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.sagemakerruntime";
const _st = "streaming";
const _xAIPV = "x-Amzn-Invoked-Production-Variant";
const n0 = "com.amazonaws.sagemakerruntime";
var BodyBlob = [0, n0, _BB, 8, 21];
var CustomAttributesHeader = [0, n0, _CAH, 8, 0];
var PartBlob = [0, n0, _PB, 8, 21];
var InternalDependencyException$ = [-3, n0, _IDE,
    { [_e]: _s, [_hE]: 530 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InternalDependencyException$, InternalDependencyException);
var InternalFailure$ = [-3, n0, _IF,
    { [_e]: _s, [_hE]: 500 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InternalFailure$, InternalFailure);
var InternalStreamFailure$ = [-3, n0, _ISF,
    { [_e]: _s },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InternalStreamFailure$, InternalStreamFailure);
var InvokeEndpointAsyncInput$ = [3, n0, _IEAI,
    0,
    [_EN, _CT, _A, _CA, _II, _IL, _RTTLS, _ITS],
    [[0, 1], [0, { [_hH]: _XASCT }], [0, { [_hH]: _XASA }], [() => CustomAttributesHeader, { [_hH]: _XASCA }], [0, { [_hH]: _XASII }], [0, { [_hH]: _XASI }], [1, { [_hH]: _XASR }], [1, { [_hH]: _XASI_ }]]
];
var InvokeEndpointAsyncOutput$ = [3, n0, _IEAO,
    0,
    [_II, _OL, _FL],
    [0, [0, { [_hH]: _XASO }], [0, { [_hH]: _XASF }]]
];
var InvokeEndpointInput$ = [3, n0, _IEI,
    0,
    [_EN, _B, _CT, _A, _CA, _TM, _TV, _TCH, _II, _EE, _ICN, _SI],
    [[0, 1], [() => BodyBlob, 16], [0, { [_hH]: _CT_ }], [0, { [_hH]: _A }], [() => CustomAttributesHeader, { [_hH]: _XASCA }], [0, { [_hH]: _XASTM }], [0, { [_hH]: _XASTV }], [0, { [_hH]: _XASTCH }], [0, { [_hH]: _XASII }], [0, { [_hH]: _XASEE }], [0, { [_hH]: _XASIC }], [0, { [_hH]: _XASSI }]]
];
var InvokeEndpointOutput$ = [3, n0, _IEO,
    0,
    [_B, _CT, _IPV, _CA, _NSI, _CSI],
    [[() => BodyBlob, 16], [0, { [_hH]: _CT_ }], [0, { [_hH]: _xAIPV }], [() => CustomAttributesHeader, { [_hH]: _XASCA }], [0, { [_hH]: _XASNSI }], [0, { [_hH]: _XASCSI }]]
];
var InvokeEndpointWithResponseStreamInput$ = [3, n0, _IEWRSI,
    0,
    [_EN, _B, _CT, _A, _CA, _TV, _TCH, _II, _ICN, _SI],
    [[0, 1], [() => BodyBlob, 16], [0, { [_hH]: _CT_ }], [0, { [_hH]: _XASA }], [() => CustomAttributesHeader, { [_hH]: _XASCA }], [0, { [_hH]: _XASTV }], [0, { [_hH]: _XASTCH }], [0, { [_hH]: _XASII }], [0, { [_hH]: _XASIC }], [0, { [_hH]: _XASSI }]]
];
var InvokeEndpointWithResponseStreamOutput$ = [3, n0, _IEWRSO,
    0,
    [_B, _CT, _IPV, _CA],
    [[() => ResponseStream$, 16], [0, { [_hH]: _XASCT }], [0, { [_hH]: _xAIPV }], [() => CustomAttributesHeader, { [_hH]: _XASCA }]]
];
var ModelError$ = [-3, n0, _ME,
    { [_e]: _c, [_hE]: 424 },
    [_M, _OSC, _OM, _LSA],
    [0, 1, 0, 0]
];
schema.TypeRegistry.for(n0).registerError(ModelError$, ModelError);
var ModelNotReadyException$ = [-3, n0, _MNRE,
    { [_aQE]: [`ModelNotReadyException`, 429], [_e]: _c, [_hE]: 429 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ModelNotReadyException$, ModelNotReadyException);
var ModelStreamError$ = [-3, n0, _MSE,
    { [_e]: _c },
    [_M, _EC],
    [0, 0]
];
schema.TypeRegistry.for(n0).registerError(ModelStreamError$, ModelStreamError);
var PayloadPart$ = [3, n0, _PP,
    0,
    [_By],
    [[() => PartBlob, { [_eP]: 1 }]]
];
var ServiceUnavailable$ = [-3, n0, _SU,
    { [_e]: _s, [_hE]: 503 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ServiceUnavailable$, ServiceUnavailable);
var ValidationError$ = [-3, n0, _VE,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ValidationError$, ValidationError);
var SageMakerRuntimeServiceException$ = [-3, _sm, "SageMakerRuntimeServiceException", 0, [], []];
schema.TypeRegistry.for(_sm).registerError(SageMakerRuntimeServiceException$, SageMakerRuntimeServiceException);
var ResponseStream$ = [4, n0, _RS,
    { [_st]: 1 },
    [_PP, _MSE, _ISF],
    [[() => PayloadPart$, 0], [() => ModelStreamError$, 0], [() => InternalStreamFailure$, 0]]
];
var InvokeEndpoint$ = [9, n0, _IE,
    { [_h]: ["POST", "/endpoints/{EndpointName}/invocations", 200] }, () => InvokeEndpointInput$, () => InvokeEndpointOutput$
];
var InvokeEndpointAsync$ = [9, n0, _IEA,
    { [_h]: ["POST", "/endpoints/{EndpointName}/async-invocations", 202] }, () => InvokeEndpointAsyncInput$, () => InvokeEndpointAsyncOutput$
];
var InvokeEndpointWithResponseStream$ = [9, n0, _IEWRS,
    { [_h]: ["POST", "/endpoints/{EndpointName}/invocations-response-stream", 200] }, () => InvokeEndpointWithResponseStreamInput$, () => InvokeEndpointWithResponseStreamOutput$
];

class InvokeEndpointAsyncCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonSageMakerRuntime", "InvokeEndpointAsync", {})
    .n("SageMakerRuntimeClient", "InvokeEndpointAsyncCommand")
    .sc(InvokeEndpointAsync$)
    .build() {
}

class InvokeEndpointCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonSageMakerRuntime", "InvokeEndpoint", {})
    .n("SageMakerRuntimeClient", "InvokeEndpointCommand")
    .sc(InvokeEndpoint$)
    .build() {
}

class InvokeEndpointWithResponseStreamCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonSageMakerRuntime", "InvokeEndpointWithResponseStream", {
    eventStream: {
        output: true,
    },
})
    .n("SageMakerRuntimeClient", "InvokeEndpointWithResponseStreamCommand")
    .sc(InvokeEndpointWithResponseStream$)
    .build() {
}

const commands = {
    InvokeEndpointCommand,
    InvokeEndpointAsyncCommand,
    InvokeEndpointWithResponseStreamCommand,
};
class SageMakerRuntime extends SageMakerRuntimeClient {
}
smithyClient.createAggregatedClient(commands, SageMakerRuntime);

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
exports.InternalDependencyException = InternalDependencyException;
exports.InternalDependencyException$ = InternalDependencyException$;
exports.InternalFailure = InternalFailure;
exports.InternalFailure$ = InternalFailure$;
exports.InternalStreamFailure = InternalStreamFailure;
exports.InternalStreamFailure$ = InternalStreamFailure$;
exports.InvokeEndpoint$ = InvokeEndpoint$;
exports.InvokeEndpointAsync$ = InvokeEndpointAsync$;
exports.InvokeEndpointAsyncCommand = InvokeEndpointAsyncCommand;
exports.InvokeEndpointAsyncInput$ = InvokeEndpointAsyncInput$;
exports.InvokeEndpointAsyncOutput$ = InvokeEndpointAsyncOutput$;
exports.InvokeEndpointCommand = InvokeEndpointCommand;
exports.InvokeEndpointInput$ = InvokeEndpointInput$;
exports.InvokeEndpointOutput$ = InvokeEndpointOutput$;
exports.InvokeEndpointWithResponseStream$ = InvokeEndpointWithResponseStream$;
exports.InvokeEndpointWithResponseStreamCommand = InvokeEndpointWithResponseStreamCommand;
exports.InvokeEndpointWithResponseStreamInput$ = InvokeEndpointWithResponseStreamInput$;
exports.InvokeEndpointWithResponseStreamOutput$ = InvokeEndpointWithResponseStreamOutput$;
exports.ModelError = ModelError;
exports.ModelError$ = ModelError$;
exports.ModelNotReadyException = ModelNotReadyException;
exports.ModelNotReadyException$ = ModelNotReadyException$;
exports.ModelStreamError = ModelStreamError;
exports.ModelStreamError$ = ModelStreamError$;
exports.PayloadPart$ = PayloadPart$;
exports.ResponseStream$ = ResponseStream$;
exports.SageMakerRuntime = SageMakerRuntime;
exports.SageMakerRuntimeClient = SageMakerRuntimeClient;
exports.SageMakerRuntimeServiceException = SageMakerRuntimeServiceException;
exports.SageMakerRuntimeServiceException$ = SageMakerRuntimeServiceException$;
exports.ServiceUnavailable = ServiceUnavailable;
exports.ServiceUnavailable$ = ServiceUnavailable$;
exports.ValidationError = ValidationError;
exports.ValidationError$ = ValidationError$;
