// @aws-sdk/client-bedrock-agent-runtime/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var eventstreamSerdeConfigResolver = require('@smithy/eventstream-serde-config-resolver');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "bedrock",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class BedrockAgentRuntimeClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = eventstreamSerdeConfigResolver.resolveEventStreamSerdeConfig(_config_6);
        const _config_8 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_7);
        const _config_9 = resolveRuntimeExtensions(_config_8, configuration?.extensions || []);
        this.config = _config_9;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultBedrockAgentRuntimeHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class BedrockAgentRuntimeServiceException extends smithyClient.ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, BedrockAgentRuntimeServiceException.prototype);
    }
}

class AccessDeniedException extends BedrockAgentRuntimeServiceException {
    name = "AccessDeniedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
    }
}
class InternalServerException extends BedrockAgentRuntimeServiceException {
    name = "InternalServerException";
    $fault = "server";
    reason;
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerException.prototype);
        this.reason = opts.reason;
    }
}
class ResourceNotFoundException extends BedrockAgentRuntimeServiceException {
    name = "ResourceNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
    }
}
class ThrottlingException extends BedrockAgentRuntimeServiceException {
    name = "ThrottlingException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottlingException.prototype);
    }
}
class ValidationException extends BedrockAgentRuntimeServiceException {
    name = "ValidationException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ValidationException.prototype);
    }
}
class BadGatewayException extends BedrockAgentRuntimeServiceException {
    name = "BadGatewayException";
    $fault = "server";
    resourceName;
    constructor(opts) {
        super({
            name: "BadGatewayException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, BadGatewayException.prototype);
        this.resourceName = opts.resourceName;
    }
}
class ConflictException extends BedrockAgentRuntimeServiceException {
    name = "ConflictException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
    }
}
class DependencyFailedException extends BedrockAgentRuntimeServiceException {
    name = "DependencyFailedException";
    $fault = "client";
    resourceName;
    constructor(opts) {
        super({
            name: "DependencyFailedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DependencyFailedException.prototype);
        this.resourceName = opts.resourceName;
    }
}
class ServiceQuotaExceededException extends BedrockAgentRuntimeServiceException {
    name = "ServiceQuotaExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
    }
}
class ModelNotReadyException extends BedrockAgentRuntimeServiceException {
    name = "ModelNotReadyException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ModelNotReadyException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ModelNotReadyException.prototype);
    }
}

const _A = "Attribution";
const _AAG = "AgentActionGroup";
const _AAGg = "AgentActionGroups";
const _ACII = "AgentCollaboratorInvocationInput";
const _ACIO = "AgentCollaboratorInvocationOutput";
const _ACIP = "AgentCollaboratorInputPayload";
const _ACM = "ApiContentMap";
const _ACOP = "AgentCollaboratorOutputPayload";
const _ACPS = "AgentCollaboratorPayloadString";
const _ADE = "AccessDeniedException";
const _AGE = "ActionGroupExecutor";
const _AGII = "ActionGroupInvocationInput";
const _AGIO = "ActionGroupInvocationOutput";
const _AGN = "ActionGroupName";
const _AGOS = "ActionGroupOutputString";
const _AII = "ApiInvocationInput";
const _AP = "ApiPath";
const _APE = "AnalyzePromptEvent";
const _APIS = "APISchema";
const _APp = "ApiParameter";
const _APpi = "ApiParameters";
const _AR = "ApiResult";
const _ARB = "ApiRequestBody";
const _AS = "AudioSegment";
const _AT = "AgentTraces";
const _BCB = "ByteContentBlob";
const _BCD = "ByteContentDoc";
const _BCF = "ByteContentFile";
const _BGE = "BadGatewayException";
const _BMC = "BedrockModelConfigurations";
const _BPT = "BasePromptTemplate";
const _BRC = "BedrockRerankingConfiguration";
const _BRMC = "BedrockRerankingModelConfiguration";
const _BSCB = "BedrockSessionContentBlocks";
const _BSCBe = "BedrockSessionContentBlock";
const _C = "Citation";
const _CB = "ContentBody";
const _CBo = "ContentBlocks";
const _CBon = "ContentBlock";
const _CC = "CollaboratorConfiguration";
const _CCa = "CallerChain";
const _CCo = "CollaboratorConfigurations";
const _CE = "CitationEvent";
const _CEo = "ConflictException";
const _CH = "ConversationHistory";
const _CI = "CollaborationInstruction";
const _CIII = "CodeInterpreterInvocationInput";
const _CIIO = "CodeInterpreterInvocationOutput";
const _CIR = "CreateInvocationRequest";
const _CIRr = "CreateInvocationResponse";
const _CIr = "CreateInvocation";
const _CM = "ContentMap";
const _CO = "CustomOrchestration";
const _COT = "CustomOrchestrationTrace";
const _COTE = "CustomOrchestrationTraceEvent";
const _CRE = "ConditionResultEvent";
const _CS = "CreateSession";
const _CSR = "CreateSessionRequest";
const _CSRr = "CreateSessionResponse";
const _Ca = "Caller";
const _Ci = "Citations";
const _Co = "Collaborator";
const _Col = "Collaborators";
const _DAM = "DeleteAgentMemory";
const _DAMR = "DeleteAgentMemoryRequest";
const _DAMRe = "DeleteAgentMemoryResponse";
const _DFE = "DependencyFailedException";
const _DS = "DeleteSession";
const _DSR = "DeleteSessionRequest";
const _DSRe = "DeleteSessionResponse";
const _ES = "ExternalSource";
const _ESGC = "ExternalSourcesGenerationConfiguration";
const _ESR = "EndSessionRequest";
const _ESRAGC = "ExternalSourcesRetrieveAndGenerateConfiguration";
const _ESRn = "EndSessionResponse";
const _ESn = "EndSession";
const _ESx = "ExternalSources";
const _F = "Function";
const _FA = "FilterAttribute";
const _FB = "FileBody";
const _FCE = "FlowCompletionEvent";
const _FD = "FunctionDefinition";
const _FEC = "FlowExecutionContent";
const _FEE = "FlowExecutionError";
const _FEEl = "FlowExecutionErrors";
const _FEElo = "FlowExecutionEvents";
const _FEElow = "FlowExecutionEvent";
const _FEIE = "FlowExecutionInputEvent";
const _FEOE = "FlowExecutionOutputEvent";
const _FES = "FlowExecutionSummary";
const _FESl = "FlowExecutionSummaries";
const _FFE = "FlowFailureEvent";
const _FFR = "FieldForReranking";
const _FFRi = "FieldsForReranking";
const _FI = "FlowInput";
const _FIC = "FlowInputContent";
const _FIF = "FlowInputField";
const _FIFl = "FlowInputFields";
const _FII = "FunctionInvocationInput";
const _FIl = "FlowInputs";
const _FMTIC = "FlowMultiTurnInputContent";
const _FMTIRE = "FlowMultiTurnInputRequestEvent";
const _FNIE = "FlowNodeInputExpression";
const _FOC = "FlowOutputContent";
const _FOE = "FlowOutputEvent";
const _FOF = "FlowOutputField";
const _FOFl = "FlowOutputFields";
const _FP = "FilePart";
const _FPu = "FunctionParameter";
const _FPun = "FunctionParameters";
const _FR = "FinalResponse";
const _FRS = "FailureReasonString";
const _FRSi = "FinalResponseString";
const _FRSl = "FlowResponseStream";
const _FRu = "FunctionResult";
const _FS = "FileSource";
const _FSu = "FunctionSchema";
const _FT = "FailureTrace";
const _FTC = "FlowTraceCondition";
const _FTCNRE = "FlowTraceConditionNodeResultEvent";
const _FTCl = "FlowTraceConditions";
const _FTDE = "FlowTraceDependencyEvent";
const _FTE = "FlowTraceEvent";
const _FTNAE = "FlowTraceNodeActionEvent";
const _FTNIC = "FlowTraceNodeInputContent";
const _FTNIE = "FlowTraceNodeInputEvent";
const _FTNIEC = "FlowTraceNodeInputExecutionChain";
const _FTNIECI = "FlowTraceNodeInputExecutionChainItem";
const _FTNIF = "FlowTraceNodeInputField";
const _FTNIFl = "FlowTraceNodeInputFields";
const _FTNIS = "FlowTraceNodeInputSource";
const _FTNOC = "FlowTraceNodeOutputContent";
const _FTNOE = "FlowTraceNodeOutputEvent";
const _FTNOF = "FlowTraceNodeOutputField";
const _FTNOFl = "FlowTraceNodeOutputFields";
const _FTNON = "FlowTraceNodeOutputNext";
const _FTNONL = "FlowTraceNodeOutputNextList";
const _FTl = "FlowTrace";
const _Fu = "Functions";
const _GA = "GuardrailAssessment";
const _GAL = "GuardrailAssessmentList";
const _GAM = "GetAgentMemory";
const _GAMR = "GetAgentMemoryRequest";
const _GAMRe = "GetAgentMemoryResponse";
const _GC = "GenerationConfiguration";
const _GCF = "GuardrailContentFilter";
const _GCFL = "GuardrailContentFilterList";
const _GCPA = "GuardrailContentPolicyAssessment";
const _GCW = "GuardrailCustomWord";
const _GCWA = "GuardrailConfigurationWithArn";
const _GCWL = "GuardrailCustomWordList";
const _GCu = "GuardrailConfiguration";
const _GE = "GuardrailEvent";
const _GEFS = "GetExecutionFlowSnapshot";
const _GEFSR = "GetExecutionFlowSnapshotRequest";
const _GEFSRe = "GetExecutionFlowSnapshotResponse";
const _GFE = "GetFlowExecution";
const _GFER = "GetFlowExecutionRequest";
const _GFERe = "GetFlowExecutionResponse";
const _GIS = "GetInvocationStep";
const _GISR = "GetInvocationStepRequest";
const _GISRe = "GetInvocationStepResponse";
const _GMW = "GuardrailManagedWord";
const _GMWL = "GuardrailManagedWordList";
const _GPEF = "GuardrailPiiEntityFilter";
const _GPEFL = "GuardrailPiiEntityFilterList";
const _GQ = "GeneratedQuery";
const _GQR = "GenerateQueryRequest";
const _GQRe = "GenerateQueryResponse";
const _GQe = "GeneratedQueries";
const _GQen = "GenerateQuery";
const _GRF = "GuardrailRegexFilter";
const _GRFL = "GuardrailRegexFilterList";
const _GRP = "GeneratedResponsePart";
const _GS = "GetSession";
const _GSIPA = "GuardrailSensitiveInformationPolicyAssessment";
const _GSR = "GetSessionRequest";
const _GSRe = "GetSessionResponse";
const _GT = "GuardrailTopic";
const _GTL = "GuardrailTopicList";
const _GTPA = "GuardrailTopicPolicyAssessment";
const _GTu = "GuardrailTrace";
const _GWPA = "GuardrailWordPolicyAssessment";
const _I = "Identifier";
const _IA = "InvokeAgent";
const _IAFP = "InlineAgentFilePart";
const _IAPP = "InlineAgentPayloadPart";
const _IAR = "InvokeAgentRequest";
const _IARCP = "InlineAgentReturnControlPayload";
const _IARS = "InlineAgentResponseStream";
const _IARn = "InvokeAgentResponse";
const _IATP = "InlineAgentTracePart";
const _IB = "ImageBlock";
const _IBMC = "InlineBedrockModelConfigurations";
const _IC = "InferenceConfig";
const _ICn = "InferenceConfiguration";
const _IF = "InputFile";
const _IFC = "ImplicitFilterConfiguration";
const _IFR = "InvokeFlowRequest";
const _IFRn = "InvokeFlowResponse";
const _IFn = "InputFiles";
const _IFnv = "InvokeFlow";
const _II = "ImageInput";
const _IIA = "InvokeInlineAgent";
const _IIAR = "InvokeInlineAgentRequest";
const _IIARn = "InvokeInlineAgentResponse";
const _IIM = "InvocationInputMember";
const _IIS = "ImageInputSource";
const _IIm = "ImageInputs";
const _IIn = "InputImage";
const _IInv = "InvocationInput";
const _IInvo = "InvocationInputs";
const _IP = "InputPrompt";
const _IRM = "InvocationResultMember";
const _IS = "InvocationStep";
const _ISE = "InternalServerException";
const _ISP = "InvocationStepPayload";
const _ISS = "InlineSessionState";
const _ISSn = "InvocationStepSummary";
const _ISSnv = "InvocationStepSummaries";
const _ISm = "ImageSource";
const _ISn = "InvocationSummary";
const _ISnv = "InvocationSummaries";
const _IT = "InputText";
const _In = "Instruction";
const _KB = "KnowledgeBase";
const _KBC = "KnowledgeBaseConfiguration";
const _KBCn = "KnowledgeBaseConfigurations";
const _KBLI = "KnowledgeBaseLookupInput";
const _KBLIS = "KnowledgeBaseLookupInputString";
const _KBLO = "KnowledgeBaseLookupOutput";
const _KBQ = "KnowledgeBaseQuery";
const _KBRAGC = "KnowledgeBaseRetrieveAndGenerateConfiguration";
const _KBRC = "KnowledgeBaseRetrievalConfiguration";
const _KBRR = "KnowledgeBaseRetrievalResult";
const _KBRRn = "KnowledgeBaseRetrievalResults";
const _KBVSC = "KnowledgeBaseVectorSearchConfiguration";
const _KBn = "KnowledgeBases";
const _LFE = "ListFlowExecutions";
const _LFEE = "ListFlowExecutionEvents";
const _LFEER = "ListFlowExecutionEventsRequest";
const _LFEERi = "ListFlowExecutionEventsResponse";
const _LFER = "ListFlowExecutionsRequest";
const _LFERi = "ListFlowExecutionsResponse";
const _LI = "ListInvocations";
const _LIR = "ListInvocationsRequest";
const _LIRi = "ListInvocationsResponse";
const _LIS = "ListInvocationSteps";
const _LISR = "ListInvocationStepsRequest";
const _LISRi = "ListInvocationStepsResponse";
const _LS = "ListSessions";
const _LSR = "ListSessionsRequest";
const _LSRi = "ListSessionsResponse";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _M = "Message";
const _MAS = "MetadataAttributeSchema";
const _MASL = "MetadataAttributeSchemaList";
const _MCFR = "MetadataConfigurationForReranking";
const _MII = "ModelInvocationInput";
const _MNRE = "ModelNotReadyException";
const _MPC = "ModelPerformanceConfiguration";
const _MSS = "MemorySessionSummary";
const _Me = "Metadata";
const _Mem = "Memories";
const _Memo = "Memory";
const _Mes = "Messages";
const _N = "Name";
const _NAE = "NodeActionEvent";
const _NDE = "NodeDependencyEvent";
const _NEC = "NodeExecutionContent";
const _NFE = "NodeFailureEvent";
const _NIE = "NodeInputEvent";
const _NIEC = "NodeInputExecutionChain";
const _NIECI = "NodeInputExecutionChainItem";
const _NIF = "NodeInputField";
const _NIFo = "NodeInputFields";
const _NIS = "NodeInputSource";
const _NOE = "NodeOutputEvent";
const _NOF = "NodeOutputField";
const _NOFo = "NodeOutputFields";
const _NON = "NodeOutputNext";
const _NONL = "NodeOutputNextList";
const _NTE = "NodeTraceElements";
const _O = "Observation";
const _OC = "OrchestrationConfiguration";
const _OE = "OrchestrationExecutor";
const _OF = "OutputFile";
const _OFu = "OutputFiles";
const _OMIO = "OrchestrationModelInvocationOutput";
const _OP = "OptimizedPrompt";
const _OPE = "OptimizedPromptEvent";
const _OPR = "OptimizePromptRequest";
const _OPRp = "OptimizePromptResponse";
const _OPS = "OptimizedPromptStream";
const _OPp = "OptimizePrompt";
const _OS = "OutputString";
const _OT = "OrchestrationTrace";
const _P = "Payload";
const _PB = "PartBody";
const _PC = "PerformanceConfiguration";
const _PCC = "PromptCreationConfigurations";
const _PCr = "PromptConfiguration";
const _PCro = "PromptConfigurations";
const _PD = "ParameterDetail";
const _PIS = "PutInvocationStep";
const _PISR = "PutInvocationStepRequest";
const _PISRu = "PutInvocationStepResponse";
const _PL = "ParameterList";
const _PM = "ParameterMap";
const _POC = "PromptOverrideConfiguration";
const _PP = "PayloadPart";
const _PPMIO = "PostProcessingModelInvocationOutput";
const _PPMIOr = "PreProcessingModelInvocationOutput";
const _PPPR = "PostProcessingParsedResponse";
const _PPPRr = "PreProcessingParsedResponse";
const _PPT = "PostProcessingTrace";
const _PPTr = "PreProcessingTrace";
const _PPr = "PropertyParameters";
const _PT = "PromptText";
const _PTr = "PromptTemplate";
const _Pa = "Parameter";
const _Par = "Parameters";
const _QGI = "QueryGenerationInput";
const _QTC = "QueryTransformationConfiguration";
const _R = "Rationale";
const _RAG = "RetrieveAndGenerate";
const _RAGC = "RetrieveAndGenerateConfiguration";
const _RAGI = "RetrieveAndGenerateInput";
const _RAGO = "RetrieveAndGenerateOutput";
const _RAGOE = "RetrieveAndGenerateOutputEvent";
const _RAGR = "RetrieveAndGenerateRequest";
const _RAGRe = "RetrieveAndGenerateResponse";
const _RAGS = "RetrieveAndGenerateStream";
const _RAGSC = "RetrieveAndGenerateSessionConfiguration";
const _RAGSR = "RetrieveAndGenerateStreamRequest";
const _RAGSRO = "RetrieveAndGenerateStreamResponseOutput";
const _RAGSRe = "RetrieveAndGenerateStreamResponse";
const _RB = "RequestBody";
const _RBe = "ResponseBody";
const _RC = "RerankingConfiguration";
const _RCB = "ReasoningContentBlock";
const _RCIR = "ReturnControlInvocationResults";
const _RCMIO = "RoutingClassifierModelInvocationOutput";
const _RCP = "ReturnControlPayload";
const _RCR = "ReturnControlResults";
const _RCT = "RoutingClassifierTrace";
const _RD = "ResourceDescription";
const _RDe = "RerankDocument";
const _RF = "RetrievalFilter";
const _RFL = "RetrievalFilterList";
const _RMSMC = "RerankingMetadataSelectiveModeConfiguration";
const _RN = "ResourceName";
const _RNFE = "ResourceNotFoundException";
const _RQ = "RerankQuery";
const _RQL = "RerankQueriesList";
const _RR = "RawResponse";
const _RRC = "RetrievalResultContent";
const _RRCC = "RetrievalResultContentColumn";
const _RRCDL = "RetrievalResultCustomDocumentLocation";
const _RRCL = "RetrievalResultConfluenceLocation";
const _RRCR = "RetrievalResultContentRow";
const _RRKDL = "RetrievalResultKendraDocumentLocation";
const _RRL = "RetrievalResultLocation";
const _RRLe = "RerankResultsList";
const _RRM = "RetrievalResultMetadata";
const _RRSL = "RetrievalResultS3Location";
const _RRSLe = "RetrievalResultSalesforceLocation";
const _RRSLet = "RetrievalResultSqlLocation";
const _RRSPL = "RetrievalResultSharePointLocation";
const _RRWL = "RetrievalResultWebLocation";
const _RRe = "RepromptResponse";
const _RRer = "RerankRequest";
const _RRera = "RerankResponse";
const _RReran = "RerankResult";
const _RRet = "RetrievedReference";
const _RRetr = "RetrieveRequest";
const _RRetri = "RetrieveResponse";
const _RRetrie = "RetrievedReferences";
const _RS = "RationaleString";
const _RSL = "RerankSourcesList";
const _RSe = "RerankSource";
const _RSes = "ResponseStream";
const _RTB = "ReasoningTextBlock";
const _RTD = "RerankTextDocument";
const _Re = "Rerank";
const _Ret = "Retrieve";
const _S = "Source";
const _SC = "SatisfiedCondition";
const _SCa = "SatisfiedConditions";
const _SCt = "StreamingConfigurations";
const _SFE = "StartFlowExecution";
const _SFER = "StartFlowExecutionRequest";
const _SFERt = "StartFlowExecutionResponse";
const _SFERto = "StopFlowExecutionRequest";
const _SFERtop = "StopFlowExecutionResponse";
const _SFEt = "StopFlowExecution";
const _SI = "S3Identifier";
const _SL = "S3Location";
const _SOD = "S3ObjectDoc";
const _SOF = "S3ObjectFile";
const _SQEE = "ServiceQuotaExceededException";
const _SS = "SessionState";
const _SSe = "SessionSummary";
const _SSes = "SessionSummaries";
const _Sp = "Span";
const _T = "Trace";
const _TC = "TransformationConfiguration";
const _TE = "ThrottlingException";
const _TEr = "TraceElements";
const _TIC = "TextInferenceConfig";
const _TKBI = "TraceKnowledgeBaseId";
const _TP = "TextPrompt";
const _TPT = "TextPromptTemplate";
const _TPr = "TracePart";
const _TR = "TagResource";
const _TRP = "TextResponsePart";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _TTSC = "TextToSqlConfiguration";
const _TTSKBC = "TextToSqlKnowledgeBaseConfiguration";
const _U = "Usage";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _US = "UpdateSession";
const _USR = "UpdateSessionRequest";
const _USRp = "UpdateSessionResponse";
const _V = "Verb";
const _VE = "ValidationException";
const _VS = "VideoSegment";
const _VSBRC = "VectorSearchBedrockRerankingConfiguration";
const _VSBRMC = "VectorSearchBedrockRerankingModelConfiguration";
const _VSRC = "VectorSearchRerankingConfiguration";
const _a = "action";
const _aA = "andAll";
const _aAA = "agentAliasArn";
const _aAI = "agentAliasId";
const _aC = "agentCollaboration";
const _aCAA = "agentCollaboratorAliasArn";
const _aCII = "agentCollaboratorInvocationInput";
const _aCIO = "agentCollaboratorInvocationOutput";
const _aCN = "agentCollaboratorName";
const _aDE = "accessDeniedException";
const _aG = "actionGroup";
const _aGE = "actionGroupExecutor";
const _aGI = "applyGuardrailInterval";
const _aGII = "actionGroupInvocationInput";
const _aGIO = "actionGroupInvocationOutput";
const _aGN = "actionGroupName";
const _aGc = "actionGroups";
const _aI = "agentId";
const _aII = "apiInvocationInput";
const _aIT = "actionInvocationType";
const _aMRF = "additionalModelRequestFields";
const _aN = "agentName";
const _aP = "apiPath";
const _aPE = "analyzePromptEvent";
const _aR = "apiResult";
const _aS = "apiSchema";
const _aT = "agentTraces";
const _aV = "agentVersion";
const _at = "attribution";
const _au = "audio";
const _b = "body";
const _bC = "byteContent";
const _bGE = "badGatewayException";
const _bMC = "bedrockModelConfigurations";
const _bPT = "basePromptTemplate";
const _bRC = "bedrockRerankingConfiguration";
const _by = "bytes";
const _c = "client";
const _cA = "createdAt";
const _cB = "contentBlocks";
const _cC = "collaboratorConfigurations";
const _cCa = "callerChain";
const _cCu = "customControl";
const _cDL = "customDocumentLocation";
const _cE = "conflictException";
const _cEKA = "customerEncryptionKeyArn";
const _cH = "conversationHistory";
const _cI = "collaboratorInstruction";
const _cIII = "codeInterpreterInvocationInput";
const _cIIO = "codeInterpreterInvocationOutput";
const _cL = "confluenceLocation";
const _cN = "collaboratorName";
const _cNRT = "conditionNodeResultTrace";
const _cNo = "conditionName";
const _cNol = "columnName";
const _cO = "customOrchestration";
const _cOT = "customOrchestrationTrace";
const _cP = "contentPolicy";
const _cR = "completionReason";
const _cRE = "conditionResultEvent";
const _cRI = "clientRequestId";
const _cS = "confirmationState";
const _cT = "contentType";
const _cV = "columnValue";
const _cW = "customWords";
const _ca = "category";
const _ch = "chunk";
const _ci = "citations";
const _cit = "citation";
const _co = "content";
const _cod = "code";
const _col = "collaborators";
const _com = "completion";
const _con = "confidence";
const _d = "description";
const _dFE = "dependencyFailedException";
const _da = "data";
const _de = "definition";
const _do = "document";
const _e = "error";
const _eA = "executionArn";
const _eAn = "endedAt";
const _eC = "errorCode";
const _eCx = "executionChain";
const _eE = "executionError";
const _eI = "executionIdentifier";
const _eIx = "executionId";
const _eKA = "encryptionKeyArn";
const _eM = "errorMessage";
const _eO = "executionOutput";
const _ePTS = "excludePreviousThinkingSteps";
const _eRA = "executionRoleArn";
const _eS = "endSession";
const _eSC = "externalSourcesConfiguration";
const _eT = "executionType";
const _eTn = "enableTrace";
const _eTnd = "endTime";
const _eTv = "eventTime";
const _eTve = "eventType";
const _eTx = "executionTimeout";
const _en = "end";
const _eq = "equals";
const _er = "errors";
const _ev = "event";
const _ex = "executor";
const _exp = "expression";
const _f = "function";
const _fAI = "flowAliasIdentifier";
const _fC = "failureCode";
const _fCE = "flowCompletionEvent";
const _fEE = "flowExecutionEvents";
const _fEN = "flowExecutionName";
const _fES = "flowExecutionSummaries";
const _fFE = "flowFailureEvent";
const _fI = "flowIdentifier";
const _fIE = "flowInputEvent";
const _fII = "functionInvocationInput";
const _fM = "foundationModel";
const _fMTIRE = "flowMultiTurnInputRequestEvent";
const _fN = "fieldName";
const _fOE = "flowOutputEvent";
const _fR = "failureReason";
const _fRi = "finalResponse";
const _fRu = "functionResult";
const _fS = "functionSchema";
const _fT = "failureTrace";
const _fTE = "flowTraceEvent";
const _fTEi = "fieldsToExclude";
const _fTI = "fieldsToInclude";
const _fV = "flowVersion";
const _fi = "files";
const _fie = "fields";
const _fil = "filters";
const _filt = "filter";
const _fo = "format";
const _fu = "functions";
const _g = "guardrail";
const _gA = "guardrailAction";
const _gC = "guardrailConfiguration";
const _gCe = "generationConfiguration";
const _gI = "guardrailId";
const _gIu = "guardrailIdentifier";
const _gRP = "generatedResponsePart";
const _gT = "greaterThan";
const _gTOE = "greaterThanOrEquals";
const _gTu = "guardrailTrace";
const _gV = "guardrailVersion";
const _h = "http";
const _hE = "httpError";
const _hH = "httpHeader";
const _hM = "httpMethod";
const _hQ = "httpQuery";
const _hSC = "httpStatusCode";
const _i = "input";
const _iA = "inputAssessments";
const _iC = "inferenceConfig";
const _iCn = "inlineContent";
const _iCnf = "inferenceConfiguration";
const _iDS = "inlineDocumentSource";
const _iFC = "implicitFilterConfiguration";
const _iFN = "inputFieldName";
const _iI = "invocationId";
const _iIn = "invocationIdentifier";
const _iInv = "invocationInputs";
const _iInvo = "invocationInput";
const _iS = "invocationStep";
const _iSE = "internalServerException";
const _iSI = "invocationStepId";
const _iSS = "inlineSessionState";
const _iSSn = "invocationStepSummaries";
const _iST = "invocationStepTime";
const _iSTTLIS = "idleSessionTTLInSeconds";
const _iSn = "invocationSummaries";
const _iT = "invocationType";
const _iTn = "inputText";
const _iTnp = "inputTokens";
const _iV = "isValid";
const _id = "identifier";
const _id_ = "id";
const _im = "images";
const _ima = "image";
const _in = "instruction";
const _in_ = "in";
const _ind = "index";
const _inp = "inputs";
const _jD = "jsonDocument";
const _k = "key";
const _kB = "knowledgeBases";
const _kBA = "knowledgeBaseArn";
const _kBC = "knowledgeBaseConfiguration";
const _kBCn = "knowledgeBaseConfigurations";
const _kBI = "knowledgeBaseId";
const _kBLI = "knowledgeBaseLookupInput";
const _kBLO = "knowledgeBaseLookupOutput";
const _kDL = "kendraDocumentLocation";
const _kKA = "kmsKeyArn";
const _l = "location";
const _lC = "listContains";
const _lT = "lessThan";
const _lTOE = "lessThanOrEquals";
const _lUA = "lastUpdatedAt";
const _la = "latency";
const _lam = "lambda";
const _m = "message";
const _mA = "modelArn";
const _mAe = "metadataAttributes";
const _mC = "modelConfiguration";
const _mCe = "memoryContents";
const _mCet = "metadataConfiguration";
const _mI = "memoryId";
const _mII = "modelInvocationInput";
const _mIO = "modelInvocationOutput";
const _mIa = "maxItems";
const _mL = "maximumLength";
const _mNRE = "modelNotReadyException";
const _mPC = "modelPerformanceConfiguration";
const _mR = "maxResults";
const _mT = "mediaType";
const _mTa = "maxTokens";
const _mTe = "memoryType";
const _mWL = "managedWordLists";
const _ma = "match";
const _me = "metadata";
const _mes = "messages";
const _mo = "mode";
const _n = "name";
const _nAE = "nodeActionEvent";
const _nAT = "nodeActionTrace";
const _nDE = "nodeDependencyEvent";
const _nDT = "nodeDependencyTrace";
const _nE = "notEquals";
const _nFE = "nodeFailureEvent";
const _nI = "notIn";
const _nIE = "nodeInputEvent";
const _nIN = "nodeInputName";
const _nIT = "nodeInputTrace";
const _nN = "nodeName";
const _nOE = "nodeOutputEvent";
const _nON = "nodeOutputName";
const _nOR = "numberOfResults";
const _nORR = "numberOfRerankedResults";
const _nOT = "nodeOutputTrace";
const _nT = "nodeType";
const _nTe = "nextToken";
const _ne = "next";
const _o = "output";
const _oA = "outputAssessments";
const _oAr = "orAll";
const _oC = "orchestrationConfiguration";
const _oFN = "outputFieldName";
const _oL = "overrideLambda";
const _oN = "operationName";
const _oP = "optimizedPrompt";
const _oPE = "optimizedPromptEvent";
const _oR = "operationRequest";
const _oRp = "operationResponse";
const _oST = "overrideSearchType";
const _oT = "orchestrationType";
const _oTTM = "operationTotalTimeMs";
const _oTr = "orchestrationTrace";
const _oTu = "outputTokens";
const _ob = "observation";
const _p = "parameters";
const _pAGS = "parentActionGroupSignature";
const _pAGSP = "parentActionGroupSignatureParams";
const _pC = "performanceConfig";
const _pCC = "promptCreationConfigurations";
const _pCM = "promptCreationMode";
const _pCTTI = "previousConversationTurnsToInclude";
const _pCr = "promptConfigurations";
const _pE = "piiEntities";
const _pM = "parserMode";
const _pOC = "promptOverrideConfiguration";
const _pPT = "preProcessingTrace";
const _pPTo = "postProcessingTrace";
const _pR = "parsedResponse";
const _pS = "promptState";
const _pSA = "promptSessionAttributes";
const _pT = "promptTemplate";
const _pTr = "promptType";
const _pa = "payload";
const _pr = "properties";
const _q = "queries";
const _qGI = "queryGenerationInput";
const _qTC = "queryTransformationConfiguration";
const _qu = "query";
const _r = "regex";
const _rA = "resourceArn";
const _rAGC = "retrieveAndGenerateConfiguration";
const _rB = "requestBody";
const _rBe = "responseBody";
const _rC = "requireConfirmation";
const _rCH = "relayConversationHistory";
const _rCIR = "returnControlInvocationResults";
const _rCP = "returnControlPayload";
const _rCR = "returnControlResults";
const _rCT = "routingClassifierTrace";
const _rCe = "retrievalConfiguration";
const _rCea = "reasoningContent";
const _rCed = "redactedContent";
const _rCer = "rerankingConfiguration";
const _rCet = "returnControl";
const _rI = "requestId";
const _rN = "resourceName";
const _rNFE = "resourceNotFoundException";
const _rQ = "retrievalQuery";
const _rR = "retrievedReferences";
const _rRa = "rawResponse";
const _rRe = "repromptResponse";
const _rRet = "retrievalResults";
const _rS = "responseState";
const _rSe = "responseStream";
const _rSel = "relevanceScore";
const _rT = "reasoningText";
const _ra = "rationale";
const _re = "regexes";
const _rea = "reason";
const _req = "required";
const _res = "results";
const _ro = "role";
const _row = "row";
const _s = "server";
const _sA = "sessionArn";
const _sAe = "sessionAttributes";
const _sAo = "sourceArn";
const _sAt = "startedAt";
const _sBN = "s3BucketName";
const _sC = "satisfiedConditions";
const _sCe = "sessionConfiguration";
const _sCt = "streamingConfigurations";
const _sCtr = "stringContains";
const _sET = "sessionExpiryTime";
const _sFR = "streamFinalResponse";
const _sI = "sessionIdentifier";
const _sIP = "sensitiveInformationPolicy";
const _sIe = "sessionId";
const _sL = "s3Location";
const _sLa = "salesforceLocation";
const _sLq = "sqlLocation";
const _sM = "sessionMetadata";
const _sMC = "selectiveModeConfiguration";
const _sMe = "selectionMode";
const _sN = "serviceName";
const _sOK = "s3ObjectKey";
const _sPL = "sharePointLocation";
const _sQEE = "serviceQuotaExceededException";
const _sS = "sessionStatus";
const _sST = "sessionStartTime";
const _sSe = "sessionState";
const _sSes = "sessionSummaries";
const _sSess = "sessionSummary";
const _sSt = "stopSequences";
const _sT = "sourceType";
const _sTt = "startTime";
const _sTu = "summaryText";
const _sU = "s3Uri";
const _sW = "startsWith";
const _s_ = "s3";
const _sc = "score";
const _si = "signature";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.bedrockagentruntime";
const _so = "sources";
const _sou = "source";
const _sp = "span";
const _sq = "sql";
const _st = "status";
const _sta = "start";
const _str = "stream";
const _stre = "streaming";
const _su = "summary";
const _t = "text";
const _tC = "transformationConfiguration";
const _tD = "textDocument";
const _tE = "traceElements";
const _tEh = "throttlingException";
const _tI = "traceId";
const _tIC = "textInferenceConfig";
const _tK = "topK";
const _tKa = "tagKeys";
const _tMI = "targetModelId";
const _tP = "topicPolicy";
const _tPT = "textPromptTemplate";
const _tPe = "textPrompt";
const _tPo = "topP";
const _tQ = "textQuery";
const _tRP = "textResponsePart";
const _tTM = "totalTimeMs";
const _tTSC = "textToSqlConfiguration";
const _ta = "tags";
const _te = "temperature";
const _ti = "timestamp";
const _to = "topics";
const _tr = "transcription";
const _tra = "trace";
const _ty = "type";
const _u = "usage";
const _uC = "useCase";
const _ur = "url";
const _uri = "uri";
const _v = "verb";
const _vE = "validationException";
const _vSC = "vectorSearchConfiguration";
const _va = "value";
const _vi = "video";
const _wL = "webLocation";
const _wP = "wordPolicy";
const _xabact = "x-amzn-bedrock-agent-content-type";
const _xabami = "x-amz-bedrock-agent-memory-id";
const _xabasi = "x-amz-bedrock-agent-session-id";
const _xabfei = "x-amz-bedrock-flow-execution-id";
const _xabkbsi = "x-amzn-bedrock-knowledge-base-session-id";
const _xasa = "x-amz-source-arn";
const n0 = "com.amazonaws.bedrockagentruntime";
var ActionGroupName = [0, n0, _AGN, 8, 0];
var ActionGroupOutputString = [0, n0, _AGOS, 8, 0];
var AgentCollaboratorPayloadString = [0, n0, _ACPS, 8, 0];
var ApiPath = [0, n0, _AP, 8, 0];
var BasePromptTemplate = [0, n0, _BPT, 8, 0];
var ByteContentBlob = [0, n0, _BCB, 8, 21];
var CollaborationInstruction = [0, n0, _CI, 8, 0];
var FailureReasonString = [0, n0, _FRS, 8, 0];
var FileBody = [0, n0, _FB, 8, 21];
var FinalResponseString = [0, n0, _FRSi, 8, 0];
var FlowNodeInputExpression = [0, n0, _FNIE, 8, 0];
var Function = [0, n0, _F, 8, 0];
var Identifier = [0, n0, _I, 8, 0];
var InputText = [0, n0, _IT, 8, 0];
var Instruction = [0, n0, _In, 8, 0];
var KnowledgeBaseLookupInputString = [0, n0, _KBLIS, 8, 0];
var Name = [0, n0, _N, 8, 0];
var OutputString = [0, n0, _OS, 8, 0];
var PartBody = [0, n0, _PB, 8, 21];
var Payload = [0, n0, _P, 8, 0];
var PromptText = [0, n0, _PT, 8, 0];
var RationaleString = [0, n0, _RS, 8, 0];
var ResourceDescription = [0, n0, _RD, 8, 0];
var ResourceName = [0, n0, _RN, 8, 0];
var Source$1 = [0, n0, _S, 8, 0];
var TextPromptTemplate = [0, n0, _TPT, 8, 0];
var TraceKnowledgeBaseId = [0, n0, _TKBI, 8, 0];
var Verb = [0, n0, _V, 8, 0];
var AccessDeniedException$ = [-3, n0, _ADE,
    { [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(AccessDeniedException$, AccessDeniedException);
var ActionGroupInvocationInput$ = [3, n0, _AGII,
    0,
    [_aGN, _v, _aP, _p, _rB, _f, _eT, _iI],
    [[() => ActionGroupName, 0], [() => Verb, 0], [() => ApiPath, 0], () => _Parameters, () => RequestBody$, [() => Function, 0], 0, 0]
];
var ActionGroupInvocationOutput$ = [3, n0, _AGIO,
    0,
    [_t, _me],
    [[() => ActionGroupOutputString, 0], [() => Metadata$, 0]]
];
var AgentActionGroup$ = [3, n0, _AAG,
    0,
    [_aGN, _d, _pAGS, _aGE, _aS, _fS, _pAGSP],
    [[() => ResourceName, 0], [() => ResourceDescription, 0], 0, () => ActionGroupExecutor$, [() => APISchema$, 0], [() => FunctionSchema$, 0], 128 | 0]
];
var AgentCollaboratorInputPayload$ = [3, n0, _ACIP,
    0,
    [_ty, _t, _rCR],
    [0, [() => AgentCollaboratorPayloadString, 0], [() => ReturnControlResults$, 0]]
];
var AgentCollaboratorInvocationInput$ = [3, n0, _ACII,
    0,
    [_aCN, _aCAA, _i],
    [0, 0, [() => AgentCollaboratorInputPayload$, 0]]
];
var AgentCollaboratorInvocationOutput$ = [3, n0, _ACIO,
    0,
    [_aCN, _aCAA, _o, _me],
    [0, 0, [() => AgentCollaboratorOutputPayload$, 0], [() => Metadata$, 0]]
];
var AgentCollaboratorOutputPayload$ = [3, n0, _ACOP,
    0,
    [_ty, _t, _rCP],
    [0, [() => AgentCollaboratorPayloadString, 0], [() => ReturnControlPayload$, 0]]
];
var AnalyzePromptEvent$ = [3, n0, _APE,
    8,
    [_m],
    [0]
];
var ApiInvocationInput$ = [3, n0, _AII,
    0,
    [_aG, _hM, _aP, _p, _rB, _aIT, _aI, _cN],
    [0, 0, [() => ApiPath, 0], () => ApiParameters, () => ApiRequestBody$, 0, 0, [() => Name, 0]]
];
var ApiParameter$ = [3, n0, _APp,
    0,
    [_n, _ty, _va],
    [0, 0, 0]
];
var ApiRequestBody$ = [3, n0, _ARB,
    0,
    [_co],
    [() => ApiContentMap]
];
var ApiResult$ = [3, n0, _AR,
    0,
    [_aG, _hM, _aP, _cS, _rS, _hSC, _rBe, _aI],
    [0, 0, [() => ApiPath, 0], 0, 0, 1, () => ResponseBody, 0]
];
var Attribution$ = [3, n0, _A,
    0,
    [_ci],
    [[() => Citations, 0]]
];
var AudioSegment$ = [3, n0, _AS,
    0,
    [_sU, _tr],
    [0, 0]
];
var BadGatewayException$ = [-3, n0, _BGE,
    { [_e]: _s, [_hE]: 502 },
    [_m, _rN],
    [0, 0]
];
schema.TypeRegistry.for(n0).registerError(BadGatewayException$, BadGatewayException);
var BedrockModelConfigurations$ = [3, n0, _BMC,
    0,
    [_pC],
    [() => PerformanceConfiguration$]
];
var BedrockRerankingConfiguration$ = [3, n0, _BRC,
    0,
    [_nOR, _mC],
    [1, () => BedrockRerankingModelConfiguration$]
];
var BedrockRerankingModelConfiguration$ = [3, n0, _BRMC,
    0,
    [_mA, _aMRF],
    [0, 128 | 15]
];
var ByteContentDoc$ = [3, n0, _BCD,
    0,
    [_id, _cT, _da],
    [[() => Identifier, 0], 0, [() => ByteContentBlob, 0]]
];
var ByteContentFile$ = [3, n0, _BCF,
    0,
    [_mT, _da],
    [0, [() => ByteContentBlob, 0]]
];
var Citation$ = [3, n0, _C,
    0,
    [_gRP, _rR],
    [[() => GeneratedResponsePart$, 0], [() => RetrievedReferences, 0]]
];
var CitationEvent$ = [3, n0, _CE,
    0,
    [_cit, _gRP, _rR],
    [[() => Citation$, 0], [() => GeneratedResponsePart$, 0], [() => RetrievedReferences, 0]]
];
var CodeInterpreterInvocationInput$ = [3, n0, _CIII,
    0,
    [_cod, _fi],
    [0, 64 | 0]
];
var CodeInterpreterInvocationOutput$ = [3, n0, _CIIO,
    0,
    [_eO, _eE, _fi, _eTx, _me],
    [0, 0, 64 | 0, 2, [() => Metadata$, 0]]
];
var Collaborator$ = [3, n0, _Co,
    0,
    [_cEKA, _fM, _in, _iSTTLIS, _aGc, _kB, _gC, _pOC, _aC, _cC, _aN],
    [0, 0, [() => Instruction, 0], 1, [() => AgentActionGroups, 0], [() => KnowledgeBases, 0], () => GuardrailConfigurationWithArn$, [() => PromptOverrideConfiguration$, 0], 0, [() => CollaboratorConfigurations, 0], [() => Name, 0]]
];
var CollaboratorConfiguration$ = [3, n0, _CC,
    0,
    [_cN, _cI, _aAA, _rCH],
    [[() => Name, 0], [() => CollaborationInstruction, 0], 0, 0]
];
var ConditionResultEvent$ = [3, n0, _CRE,
    8,
    [_nN, _ti, _sC],
    [0, 5, [() => SatisfiedConditions, 0]]
];
var ConflictException$ = [-3, n0, _CEo,
    { [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ConflictException$, ConflictException);
var ContentBody$ = [3, n0, _CB,
    0,
    [_b, _im],
    [0, () => ImageInputs]
];
var ConversationHistory$ = [3, n0, _CH,
    0,
    [_mes],
    [[() => Messages, 0]]
];
var CreateInvocationRequest$ = [3, n0, _CIR,
    0,
    [_iI, _d, _sI],
    [0, 0, [0, 1]]
];
var CreateInvocationResponse$ = [3, n0, _CIRr,
    0,
    [_sIe, _iI, _cA],
    [0, 0, 5]
];
var CreateSessionRequest$ = [3, n0, _CSR,
    0,
    [_sM, _eKA, _ta],
    [128 | 0, 0, 128 | 0]
];
var CreateSessionResponse$ = [3, n0, _CSRr,
    0,
    [_sIe, _sA, _sS, _cA],
    [0, 0, 0, 5]
];
var CustomOrchestration$ = [3, n0, _CO,
    0,
    [_ex],
    [() => OrchestrationExecutor$]
];
var CustomOrchestrationTrace$ = [3, n0, _COT,
    8,
    [_tI, _ev],
    [0, [() => CustomOrchestrationTraceEvent$, 0]]
];
var CustomOrchestrationTraceEvent$ = [3, n0, _COTE,
    8,
    [_t],
    [0]
];
var DeleteAgentMemoryRequest$ = [3, n0, _DAMR,
    0,
    [_aI, _aAI, _mI, _sIe],
    [[0, 1], [0, 1], [0, { [_hQ]: _mI }], [0, { [_hQ]: _sIe }]]
];
var DeleteAgentMemoryResponse$ = [3, n0, _DAMRe,
    0,
    [],
    []
];
var DeleteSessionRequest$ = [3, n0, _DSR,
    0,
    [_sI],
    [[0, 1]]
];
var DeleteSessionResponse$ = [3, n0, _DSRe,
    0,
    [],
    []
];
var DependencyFailedException$ = [-3, n0, _DFE,
    { [_e]: _c, [_hE]: 424 },
    [_m, _rN],
    [0, 0]
];
schema.TypeRegistry.for(n0).registerError(DependencyFailedException$, DependencyFailedException);
var EndSessionRequest$ = [3, n0, _ESR,
    0,
    [_sI],
    [[0, 1]]
];
var EndSessionResponse$ = [3, n0, _ESRn,
    0,
    [_sIe, _sA, _sS],
    [0, 0, 0]
];
var ExternalSource$ = [3, n0, _ES,
    0,
    [_sT, _sL, _bC],
    [0, () => S3ObjectDoc$, [() => ByteContentDoc$, 0]]
];
var ExternalSourcesGenerationConfiguration$ = [3, n0, _ESGC,
    0,
    [_pT, _gC, _iC, _aMRF, _pC],
    [[() => PromptTemplate$, 0], () => GuardrailConfiguration$, () => InferenceConfig$, 128 | 15, () => PerformanceConfiguration$]
];
var ExternalSourcesRetrieveAndGenerateConfiguration$ = [3, n0, _ESRAGC,
    0,
    [_mA, _so, _gCe],
    [0, [() => ExternalSources, 0], [() => ExternalSourcesGenerationConfiguration$, 0]]
];
var FailureTrace$ = [3, n0, _FT,
    8,
    [_tI, _fR, _fC, _me],
    [0, [() => FailureReasonString, 0], 1, [() => Metadata$, 0]]
];
var FieldForReranking$ = [3, n0, _FFR,
    0,
    [_fN],
    [0]
];
var FilePart$ = [3, n0, _FP,
    0,
    [_fi],
    [[() => OutputFiles, 0]]
];
var FileSource$ = [3, n0, _FS,
    0,
    [_sT, _sL, _bC],
    [0, () => S3ObjectFile$, [() => ByteContentFile$, 0]]
];
var FilterAttribute$ = [3, n0, _FA,
    0,
    [_k, _va],
    [0, 15]
];
var FinalResponse$ = [3, n0, _FR,
    0,
    [_t, _me],
    [[() => FinalResponseString, 0], [() => Metadata$, 0]]
];
var FlowCompletionEvent$ = [3, n0, _FCE,
    8,
    [_cR],
    [0]
];
var FlowExecutionError$ = [3, n0, _FEE,
    0,
    [_nN, _e, _m],
    [0, 0, 0]
];
var FlowExecutionInputEvent$ = [3, n0, _FEIE,
    8,
    [_nN, _ti, _fie],
    [0, 5, [() => FlowInputFields, 0]]
];
var FlowExecutionOutputEvent$ = [3, n0, _FEOE,
    8,
    [_nN, _ti, _fie],
    [0, 5, [() => FlowOutputFields, 0]]
];
var FlowExecutionSummary$ = [3, n0, _FES,
    0,
    [_eA, _fAI, _fI, _fV, _st, _cA, _eAn],
    [0, 0, 0, 0, 0, 5, 5]
];
var FlowFailureEvent$ = [3, n0, _FFE,
    8,
    [_ti, _eC, _eM],
    [5, 0, 0]
];
var FlowInput$ = [3, n0, _FI,
    0,
    [_nN, _nON, _co, _nIN],
    [0, 0, [() => FlowInputContent$, 0], 0]
];
var FlowInputField$ = [3, n0, _FIF,
    8,
    [_n, _co],
    [0, [() => FlowExecutionContent$, 0]]
];
var FlowMultiTurnInputRequestEvent$ = [3, n0, _FMTIRE,
    8,
    [_nN, _nT, _co],
    [0, 0, () => FlowMultiTurnInputContent$]
];
var FlowOutputEvent$ = [3, n0, _FOE,
    8,
    [_nN, _nT, _co],
    [0, 0, () => FlowOutputContent$]
];
var FlowOutputField$ = [3, n0, _FOF,
    8,
    [_n, _co],
    [0, [() => FlowExecutionContent$, 0]]
];
var FlowTraceCondition$ = [3, n0, _FTC,
    8,
    [_cNo],
    [0]
];
var FlowTraceConditionNodeResultEvent$ = [3, n0, _FTCNRE,
    8,
    [_nN, _ti, _sC],
    [0, 5, [() => FlowTraceConditions, 0]]
];
var FlowTraceDependencyEvent$ = [3, n0, _FTDE,
    8,
    [_nN, _ti, _tE],
    [0, 5, [() => TraceElements$, 0]]
];
var FlowTraceEvent$ = [3, n0, _FTE,
    0,
    [_tra],
    [[() => FlowTrace$, 0]]
];
var FlowTraceNodeActionEvent$ = [3, n0, _FTNAE,
    8,
    [_nN, _ti, _rI, _sN, _oN, _oR, _oRp],
    [0, 5, 0, 0, 0, 15, 15]
];
var FlowTraceNodeInputEvent$ = [3, n0, _FTNIE,
    8,
    [_nN, _ti, _fie],
    [0, 5, [() => FlowTraceNodeInputFields, 0]]
];
var FlowTraceNodeInputExecutionChainItem$ = [3, n0, _FTNIECI,
    8,
    [_nN, _ind, _ty],
    [0, 1, 0]
];
var FlowTraceNodeInputField$ = [3, n0, _FTNIF,
    8,
    [_nIN, _co, _sou, _ty, _ca, _eCx],
    [0, [() => FlowTraceNodeInputContent$, 0], [() => FlowTraceNodeInputSource$, 0], 0, 0, [() => FlowTraceNodeInputExecutionChain, 0]]
];
var FlowTraceNodeInputSource$ = [3, n0, _FTNIS,
    8,
    [_nN, _oFN, _exp],
    [0, 0, [() => FlowNodeInputExpression, 0]]
];
var FlowTraceNodeOutputEvent$ = [3, n0, _FTNOE,
    8,
    [_nN, _ti, _fie],
    [0, 5, [() => FlowTraceNodeOutputFields, 0]]
];
var FlowTraceNodeOutputField$ = [3, n0, _FTNOF,
    8,
    [_nON, _co, _ne, _ty],
    [0, () => FlowTraceNodeOutputContent$, [() => FlowTraceNodeOutputNextList, 0], 0]
];
var FlowTraceNodeOutputNext$ = [3, n0, _FTNON,
    8,
    [_nN, _iFN],
    [0, 0]
];
var FunctionDefinition$ = [3, n0, _FD,
    0,
    [_n, _d, _p, _rC],
    [[() => ResourceName, 0], 0, () => ParameterMap, 0]
];
var FunctionInvocationInput$ = [3, n0, _FII,
    0,
    [_aG, _p, _f, _aIT, _aI, _cN],
    [0, () => FunctionParameters, 0, 0, 0, [() => Name, 0]]
];
var FunctionParameter$ = [3, n0, _FPu,
    0,
    [_n, _ty, _va],
    [0, 0, 0]
];
var FunctionResult$ = [3, n0, _FRu,
    0,
    [_aG, _cS, _f, _rBe, _rS, _aI],
    [0, 0, 0, () => ResponseBody, 0, 0]
];
var GeneratedQuery$ = [3, n0, _GQ,
    8,
    [_ty, _sq],
    [0, 0]
];
var GeneratedResponsePart$ = [3, n0, _GRP,
    0,
    [_tRP],
    [[() => TextResponsePart$, 0]]
];
var GenerateQueryRequest$ = [3, n0, _GQR,
    0,
    [_qGI, _tC],
    [[() => QueryGenerationInput$, 0], () => TransformationConfiguration$]
];
var GenerateQueryResponse$ = [3, n0, _GQRe,
    0,
    [_q],
    [[() => GeneratedQueries, 0]]
];
var GenerationConfiguration$ = [3, n0, _GC,
    0,
    [_pT, _gC, _iC, _aMRF, _pC],
    [[() => PromptTemplate$, 0], () => GuardrailConfiguration$, () => InferenceConfig$, 128 | 15, () => PerformanceConfiguration$]
];
var GetAgentMemoryRequest$ = [3, n0, _GAMR,
    0,
    [_nTe, _mIa, _aI, _aAI, _mTe, _mI],
    [[0, { [_hQ]: _nTe }], [1, { [_hQ]: _mIa }], [0, 1], [0, 1], [0, { [_hQ]: _mTe }], [0, { [_hQ]: _mI }]]
];
var GetAgentMemoryResponse$ = [3, n0, _GAMRe,
    0,
    [_nTe, _mCe],
    [0, () => Memories]
];
var GetExecutionFlowSnapshotRequest$ = [3, n0, _GEFSR,
    0,
    [_fI, _fAI, _eI],
    [[0, 1], [0, 1], [0, 1]]
];
var GetExecutionFlowSnapshotResponse$ = [3, n0, _GEFSRe,
    0,
    [_fI, _fAI, _fV, _eRA, _de, _cEKA],
    [0, 0, 0, 0, 0, 0]
];
var GetFlowExecutionRequest$ = [3, n0, _GFER,
    0,
    [_fI, _fAI, _eI],
    [[0, 1], [0, 1], [0, 1]]
];
var GetFlowExecutionResponse$ = [3, n0, _GFERe,
    0,
    [_eA, _st, _sAt, _eAn, _er, _fAI, _fI, _fV],
    [0, 0, 5, 5, () => FlowExecutionErrors, 0, 0, 0]
];
var GetInvocationStepRequest$ = [3, n0, _GISR,
    0,
    [_iIn, _iSI, _sI],
    [0, [0, 1], [0, 1]]
];
var GetInvocationStepResponse$ = [3, n0, _GISRe,
    0,
    [_iS],
    [[() => InvocationStep$, 0]]
];
var GetSessionRequest$ = [3, n0, _GSR,
    0,
    [_sI],
    [[0, 1]]
];
var GetSessionResponse$ = [3, n0, _GSRe,
    0,
    [_sIe, _sA, _sS, _cA, _lUA, _sM, _eKA],
    [0, 0, 0, 5, 5, 128 | 0, 0]
];
var GuardrailAssessment$ = [3, n0, _GA,
    8,
    [_tP, _cP, _wP, _sIP],
    [[() => GuardrailTopicPolicyAssessment$, 0], [() => GuardrailContentPolicyAssessment$, 0], [() => GuardrailWordPolicyAssessment$, 0], [() => GuardrailSensitiveInformationPolicyAssessment$, 0]]
];
var GuardrailConfiguration$ = [3, n0, _GCu,
    0,
    [_gI, _gV],
    [0, 0]
];
var GuardrailConfigurationWithArn$ = [3, n0, _GCWA,
    0,
    [_gIu, _gV],
    [0, 0]
];
var GuardrailContentFilter$ = [3, n0, _GCF,
    8,
    [_ty, _con, _a],
    [0, 0, 0]
];
var GuardrailContentPolicyAssessment$ = [3, n0, _GCPA,
    8,
    [_fil],
    [[() => GuardrailContentFilterList, 0]]
];
var GuardrailCustomWord$ = [3, n0, _GCW,
    8,
    [_ma, _a],
    [0, 0]
];
var GuardrailEvent$ = [3, n0, _GE,
    0,
    [_a],
    [0]
];
var GuardrailManagedWord$ = [3, n0, _GMW,
    8,
    [_ma, _ty, _a],
    [0, 0, 0]
];
var GuardrailPiiEntityFilter$ = [3, n0, _GPEF,
    8,
    [_ty, _ma, _a],
    [0, 0, 0]
];
var GuardrailRegexFilter$ = [3, n0, _GRF,
    8,
    [_n, _r, _ma, _a],
    [0, 0, 0, 0]
];
var GuardrailSensitiveInformationPolicyAssessment$ = [3, n0, _GSIPA,
    8,
    [_pE, _re],
    [[() => GuardrailPiiEntityFilterList, 0], [() => GuardrailRegexFilterList, 0]]
];
var GuardrailTopic$ = [3, n0, _GT,
    8,
    [_n, _ty, _a],
    [0, 0, 0]
];
var GuardrailTopicPolicyAssessment$ = [3, n0, _GTPA,
    8,
    [_to],
    [[() => GuardrailTopicList, 0]]
];
var GuardrailTrace$ = [3, n0, _GTu,
    8,
    [_a, _tI, _iA, _oA, _me],
    [0, 0, [() => GuardrailAssessmentList, 0], [() => GuardrailAssessmentList, 0], [() => Metadata$, 0]]
];
var GuardrailWordPolicyAssessment$ = [3, n0, _GWPA,
    8,
    [_cW, _mWL],
    [[() => GuardrailCustomWordList, 0], [() => GuardrailManagedWordList, 0]]
];
var ImageBlock$ = [3, n0, _IB,
    0,
    [_fo, _sou],
    [0, () => ImageSource$]
];
var ImageInput$ = [3, n0, _II,
    0,
    [_fo, _sou],
    [0, () => ImageInputSource$]
];
var ImplicitFilterConfiguration$ = [3, n0, _IFC,
    0,
    [_mAe, _mA],
    [[() => MetadataAttributeSchemaList, 0], 0]
];
var InferenceConfig$ = [3, n0, _IC,
    0,
    [_tIC],
    [() => TextInferenceConfig$]
];
var InferenceConfiguration$ = [3, n0, _ICn,
    0,
    [_te, _tPo, _tK, _mL, _sSt],
    [1, 1, 1, 1, 64 | 0]
];
var InlineAgentFilePart$ = [3, n0, _IAFP,
    0,
    [_fi],
    [[() => OutputFiles, 0]]
];
var InlineAgentPayloadPart$ = [3, n0, _IAPP,
    8,
    [_by, _at],
    [[() => PartBody, 0], [() => Attribution$, 0]]
];
var InlineAgentReturnControlPayload$ = [3, n0, _IARCP,
    8,
    [_iInv, _iI],
    [[() => InvocationInputs, 0], 0]
];
var InlineAgentTracePart$ = [3, n0, _IATP,
    8,
    [_sIe, _tra, _cCa, _eTv, _cN],
    [0, [() => Trace$, 0], () => CallerChain, 5, [() => Name, 0]]
];
var InlineBedrockModelConfigurations$ = [3, n0, _IBMC,
    0,
    [_pC],
    [() => PerformanceConfiguration$]
];
var InlineSessionState$ = [3, n0, _ISS,
    0,
    [_sAe, _pSA, _rCIR, _iI, _fi, _cH],
    [128 | 0, 128 | 0, [() => ReturnControlInvocationResults, 0], 0, [() => InputFiles, 0], [() => ConversationHistory$, 0]]
];
var InputFile$ = [3, n0, _IF,
    0,
    [_n, _sou, _uC],
    [0, [() => FileSource$, 0], 0]
];
var InputImage$ = [3, n0, _IIn,
    0,
    [_fo, _iCn],
    [0, [() => ByteContentBlob, 0]]
];
var InternalServerException$ = [-3, n0, _ISE,
    { [_e]: _s, [_hE]: 500 },
    [_m, _rea],
    [0, 0]
];
schema.TypeRegistry.for(n0).registerError(InternalServerException$, InternalServerException);
var InvocationInput$ = [3, n0, _IInv,
    8,
    [_tI, _iT, _aGII, _kBLI, _cIII, _aCII],
    [0, 0, [() => ActionGroupInvocationInput$, 0], [() => KnowledgeBaseLookupInput$, 0], () => CodeInterpreterInvocationInput$, [() => AgentCollaboratorInvocationInput$, 0]]
];
var InvocationStep$ = [3, n0, _IS,
    0,
    [_sIe, _iI, _iSI, _iST, _pa],
    [0, 0, 0, 5, [() => InvocationStepPayload$, 0]]
];
var InvocationStepSummary$ = [3, n0, _ISSn,
    0,
    [_sIe, _iI, _iSI, _iST],
    [0, 0, 0, 5]
];
var InvocationSummary$ = [3, n0, _ISn,
    0,
    [_sIe, _iI, _cA],
    [0, 0, 5]
];
var InvokeAgentRequest$ = [3, n0, _IAR,
    0,
    [_sSe, _aI, _aAI, _sIe, _eS, _eTn, _iTn, _mI, _bMC, _sCt, _pCC, _sAo],
    [[() => SessionState$, 0], [0, 1], [0, 1], [0, 1], 2, 2, [() => InputText, 0], 0, () => BedrockModelConfigurations$, () => StreamingConfigurations$, () => PromptCreationConfigurations$, [0, { [_hH]: _xasa }]]
];
var InvokeAgentResponse$ = [3, n0, _IARn,
    0,
    [_com, _cT, _sIe, _mI],
    [[() => ResponseStream$, 16], [0, { [_hH]: _xabact }], [0, { [_hH]: _xabasi }], [0, { [_hH]: _xabami }]]
];
var InvokeFlowRequest$ = [3, n0, _IFR,
    0,
    [_fI, _fAI, _inp, _eTn, _mPC, _eIx],
    [[0, 1], [0, 1], [() => FlowInputs, 0], 2, () => ModelPerformanceConfiguration$, 0]
];
var InvokeFlowResponse$ = [3, n0, _IFRn,
    0,
    [_rSe, _eIx],
    [[() => FlowResponseStream$, 16], [0, { [_hH]: _xabfei }]]
];
var InvokeInlineAgentRequest$ = [3, n0, _IIAR,
    0,
    [_cEKA, _fM, _in, _iSTTLIS, _aGc, _kB, _gC, _pOC, _aC, _cC, _aN, _sIe, _eS, _eTn, _iTn, _sCt, _pCC, _iSS, _col, _bMC, _oT, _cO],
    [0, 0, [() => Instruction, 0], 1, [() => AgentActionGroups, 0], [() => KnowledgeBases, 0], () => GuardrailConfigurationWithArn$, [() => PromptOverrideConfiguration$, 0], 0, [() => CollaboratorConfigurations, 0], [() => Name, 0], [0, 1], 2, 2, [() => InputText, 0], () => StreamingConfigurations$, () => PromptCreationConfigurations$, [() => InlineSessionState$, 0], [() => Collaborators, 0], () => InlineBedrockModelConfigurations$, 0, () => CustomOrchestration$]
];
var InvokeInlineAgentResponse$ = [3, n0, _IIARn,
    0,
    [_com, _cT, _sIe],
    [[() => InlineAgentResponseStream$, 16], [0, { [_hH]: _xabact }], [0, { [_hH]: _xabasi }]]
];
var KnowledgeBase$ = [3, n0, _KB,
    0,
    [_kBI, _d, _rCe],
    [0, [() => ResourceDescription, 0], [() => KnowledgeBaseRetrievalConfiguration$, 0]]
];
var KnowledgeBaseConfiguration$ = [3, n0, _KBC,
    0,
    [_kBI, _rCe],
    [0, [() => KnowledgeBaseRetrievalConfiguration$, 0]]
];
var KnowledgeBaseLookupInput$ = [3, n0, _KBLI,
    0,
    [_t, _kBI],
    [[() => KnowledgeBaseLookupInputString, 0], [() => TraceKnowledgeBaseId, 0]]
];
var KnowledgeBaseLookupOutput$ = [3, n0, _KBLO,
    0,
    [_rR, _me],
    [[() => RetrievedReferences, 0], [() => Metadata$, 0]]
];
var KnowledgeBaseQuery$ = [3, n0, _KBQ,
    8,
    [_ty, _t, _ima],
    [0, 0, [() => InputImage$, 0]]
];
var KnowledgeBaseRetrievalConfiguration$ = [3, n0, _KBRC,
    0,
    [_vSC],
    [[() => KnowledgeBaseVectorSearchConfiguration$, 0]]
];
var KnowledgeBaseRetrievalResult$ = [3, n0, _KBRR,
    0,
    [_co, _l, _sc, _me],
    [[() => RetrievalResultContent$, 0], [() => RetrievalResultLocation$, 0], 1, [() => RetrievalResultMetadata, 0]]
];
var KnowledgeBaseRetrieveAndGenerateConfiguration$ = [3, n0, _KBRAGC,
    0,
    [_kBI, _mA, _rCe, _gCe, _oC],
    [0, 0, [() => KnowledgeBaseRetrievalConfiguration$, 0], [() => GenerationConfiguration$, 0], [() => OrchestrationConfiguration$, 0]]
];
var KnowledgeBaseVectorSearchConfiguration$ = [3, n0, _KBVSC,
    0,
    [_nOR, _oST, _filt, _rCer, _iFC],
    [1, 0, [() => RetrievalFilter$, 0], [() => VectorSearchRerankingConfiguration$, 0], [() => ImplicitFilterConfiguration$, 0]]
];
var ListFlowExecutionEventsRequest$ = [3, n0, _LFEER,
    0,
    [_fI, _fAI, _eI, _mR, _nTe, _eTve],
    [[0, 1], [0, 1], [0, 1], [1, { [_hQ]: _mR }], [0, { [_hQ]: _nTe }], [0, { [_hQ]: _eTve }]]
];
var ListFlowExecutionEventsResponse$ = [3, n0, _LFEERi,
    0,
    [_fEE, _nTe],
    [[() => FlowExecutionEvents, 0], 0]
];
var ListFlowExecutionsRequest$ = [3, n0, _LFER,
    0,
    [_fI, _fAI, _mR, _nTe],
    [[0, 1], [0, { [_hQ]: _fAI }], [1, { [_hQ]: _mR }], [0, { [_hQ]: _nTe }]]
];
var ListFlowExecutionsResponse$ = [3, n0, _LFERi,
    0,
    [_fES, _nTe],
    [() => FlowExecutionSummaries, 0]
];
var ListInvocationsRequest$ = [3, n0, _LIR,
    0,
    [_nTe, _mR, _sI],
    [[0, { [_hQ]: _nTe }], [1, { [_hQ]: _mR }], [0, 1]]
];
var ListInvocationsResponse$ = [3, n0, _LIRi,
    0,
    [_iSn, _nTe],
    [() => InvocationSummaries, 0]
];
var ListInvocationStepsRequest$ = [3, n0, _LISR,
    0,
    [_iIn, _nTe, _mR, _sI],
    [0, [0, { [_hQ]: _nTe }], [1, { [_hQ]: _mR }], [0, 1]]
];
var ListInvocationStepsResponse$ = [3, n0, _LISRi,
    0,
    [_iSSn, _nTe],
    [() => InvocationStepSummaries, 0]
];
var ListSessionsRequest$ = [3, n0, _LSR,
    0,
    [_mR, _nTe],
    [[1, { [_hQ]: _mR }], [0, { [_hQ]: _nTe }]]
];
var ListSessionsResponse$ = [3, n0, _LSRi,
    0,
    [_sSes, _nTe],
    [() => SessionSummaries, 0]
];
var ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_rA],
    [[0, 1]]
];
var ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_ta],
    [128 | 0]
];
var MemorySessionSummary$ = [3, n0, _MSS,
    0,
    [_mI, _sIe, _sST, _sET, _sTu],
    [0, 0, 5, 5, 0]
];
var Message$ = [3, n0, _M,
    0,
    [_ro, _co],
    [0, [() => ContentBlocks, 0]]
];
var Metadata$ = [3, n0, _Me,
    8,
    [_sTt, _eTnd, _tTM, _oTTM, _cRI, _u],
    [5, 5, 1, 1, 0, [() => Usage$, 0]]
];
var MetadataAttributeSchema$ = [3, n0, _MAS,
    8,
    [_k, _ty, _d],
    [0, 0, 0]
];
var MetadataConfigurationForReranking$ = [3, n0, _MCFR,
    0,
    [_sMe, _sMC],
    [0, [() => RerankingMetadataSelectiveModeConfiguration$, 0]]
];
var ModelInvocationInput$ = [3, n0, _MII,
    8,
    [_tI, _t, _ty, _oL, _pCM, _iCnf, _pM, _fM],
    [0, [() => PromptText, 0], 0, 0, 0, () => InferenceConfiguration$, 0, 0]
];
var ModelNotReadyException$ = [-3, n0, _MNRE,
    { [_e]: _c, [_hE]: 424 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ModelNotReadyException$, ModelNotReadyException);
var ModelPerformanceConfiguration$ = [3, n0, _MPC,
    0,
    [_pC],
    [() => PerformanceConfiguration$]
];
var NodeActionEvent$ = [3, n0, _NAE,
    8,
    [_nN, _ti, _rI, _sN, _oN, _oR, _oRp],
    [0, 5, 0, 0, 0, 15, 15]
];
var NodeDependencyEvent$ = [3, n0, _NDE,
    8,
    [_nN, _ti, _tE],
    [0, 5, [() => NodeTraceElements$, 0]]
];
var NodeFailureEvent$ = [3, n0, _NFE,
    8,
    [_nN, _ti, _eC, _eM],
    [0, 5, 0, 0]
];
var NodeInputEvent$ = [3, n0, _NIE,
    8,
    [_nN, _ti, _fie],
    [0, 5, [() => NodeInputFields, 0]]
];
var NodeInputExecutionChainItem$ = [3, n0, _NIECI,
    0,
    [_nN, _ind, _ty],
    [0, 1, 0]
];
var NodeInputField$ = [3, n0, _NIF,
    8,
    [_n, _co, _sou, _ty, _ca, _eCx],
    [0, [() => NodeExecutionContent$, 0], [() => NodeInputSource$, 0], 0, 0, () => NodeInputExecutionChain]
];
var NodeInputSource$ = [3, n0, _NIS,
    0,
    [_nN, _oFN, _exp],
    [0, 0, [() => FlowNodeInputExpression, 0]]
];
var NodeOutputEvent$ = [3, n0, _NOE,
    8,
    [_nN, _ti, _fie],
    [0, 5, [() => NodeOutputFields, 0]]
];
var NodeOutputField$ = [3, n0, _NOF,
    8,
    [_n, _co, _ne, _ty],
    [0, [() => NodeExecutionContent$, 0], [() => NodeOutputNextList, 0], 0]
];
var NodeOutputNext$ = [3, n0, _NON,
    8,
    [_nN, _iFN],
    [0, 0]
];
var Observation$ = [3, n0, _O,
    8,
    [_tI, _ty, _aGIO, _aCIO, _kBLO, _fRi, _rRe, _cIIO],
    [0, 0, [() => ActionGroupInvocationOutput$, 0], [() => AgentCollaboratorInvocationOutput$, 0], [() => KnowledgeBaseLookupOutput$, 0], [() => FinalResponse$, 0], [() => RepromptResponse$, 0], [() => CodeInterpreterInvocationOutput$, 0]]
];
var OptimizedPromptEvent$ = [3, n0, _OPE,
    8,
    [_oP],
    [[() => OptimizedPrompt$, 0]]
];
var OptimizePromptRequest$ = [3, n0, _OPR,
    0,
    [_i, _tMI],
    [[() => InputPrompt$, 0], 0]
];
var OptimizePromptResponse$ = [3, n0, _OPRp,
    0,
    [_oP],
    [[() => OptimizedPromptStream$, 16]]
];
var OrchestrationConfiguration$ = [3, n0, _OC,
    0,
    [_pT, _iC, _aMRF, _qTC, _pC],
    [[() => PromptTemplate$, 0], () => InferenceConfig$, 128 | 15, () => QueryTransformationConfiguration$, () => PerformanceConfiguration$]
];
var OrchestrationModelInvocationOutput$ = [3, n0, _OMIO,
    8,
    [_tI, _rRa, _me, _rCea],
    [0, [() => RawResponse$, 0], [() => Metadata$, 0], [() => ReasoningContentBlock$, 0]]
];
var OutputFile$ = [3, n0, _OF,
    8,
    [_n, _ty, _by],
    [0, 0, [() => FileBody, 0]]
];
var Parameter$ = [3, n0, _Pa,
    0,
    [_n, _ty, _va],
    [0, 0, 0]
];
var ParameterDetail$ = [3, n0, _PD,
    0,
    [_d, _ty, _req],
    [0, 0, 2]
];
var PayloadPart$ = [3, n0, _PP,
    8,
    [_by, _at],
    [[() => PartBody, 0], [() => Attribution$, 0]]
];
var PerformanceConfiguration$ = [3, n0, _PC,
    0,
    [_la],
    [0]
];
var PostProcessingModelInvocationOutput$ = [3, n0, _PPMIO,
    8,
    [_tI, _pR, _rRa, _me, _rCea],
    [0, [() => PostProcessingParsedResponse$, 0], [() => RawResponse$, 0], [() => Metadata$, 0], [() => ReasoningContentBlock$, 0]]
];
var PostProcessingParsedResponse$ = [3, n0, _PPPR,
    8,
    [_t],
    [[() => OutputString, 0]]
];
var PreProcessingModelInvocationOutput$ = [3, n0, _PPMIOr,
    8,
    [_tI, _pR, _rRa, _me, _rCea],
    [0, [() => PreProcessingParsedResponse$, 0], [() => RawResponse$, 0], [() => Metadata$, 0], [() => ReasoningContentBlock$, 0]]
];
var PreProcessingParsedResponse$ = [3, n0, _PPPRr,
    8,
    [_ra, _iV],
    [[() => RationaleString, 0], 2]
];
var PromptConfiguration$ = [3, n0, _PCr,
    0,
    [_pTr, _pCM, _pS, _bPT, _iCnf, _pM, _fM, _aMRF],
    [0, 0, 0, [() => BasePromptTemplate, 0], () => InferenceConfiguration$, 0, 0, 15]
];
var PromptCreationConfigurations$ = [3, n0, _PCC,
    0,
    [_pCTTI, _ePTS],
    [1, 2]
];
var PromptOverrideConfiguration$ = [3, n0, _POC,
    8,
    [_pCr, _oL],
    [[() => PromptConfigurations, 0], 0]
];
var PromptTemplate$ = [3, n0, _PTr,
    0,
    [_tPT],
    [[() => TextPromptTemplate, 0]]
];
var PropertyParameters$ = [3, n0, _PPr,
    0,
    [_pr],
    [() => ParameterList]
];
var PutInvocationStepRequest$ = [3, n0, _PISR,
    0,
    [_sI, _iIn, _iST, _pa, _iSI],
    [[0, 1], 0, 5, [() => InvocationStepPayload$, 0], 0]
];
var PutInvocationStepResponse$ = [3, n0, _PISRu,
    0,
    [_iSI],
    [0]
];
var QueryGenerationInput$ = [3, n0, _QGI,
    8,
    [_ty, _t],
    [0, 0]
];
var QueryTransformationConfiguration$ = [3, n0, _QTC,
    0,
    [_ty],
    [0]
];
var Rationale$ = [3, n0, _R,
    8,
    [_tI, _t],
    [0, [() => RationaleString, 0]]
];
var RawResponse$ = [3, n0, _RR,
    8,
    [_co],
    [0]
];
var ReasoningTextBlock$ = [3, n0, _RTB,
    8,
    [_t, _si],
    [0, 0]
];
var RepromptResponse$ = [3, n0, _RRe,
    8,
    [_t, _sou],
    [0, [() => Source$1, 0]]
];
var RequestBody$ = [3, n0, _RB,
    0,
    [_co],
    [() => ContentMap]
];
var RerankDocument$ = [3, n0, _RDe,
    8,
    [_ty, _tD, _jD],
    [0, [() => RerankTextDocument$, 0], 15]
];
var RerankingConfiguration$ = [3, n0, _RC,
    0,
    [_ty, _bRC],
    [0, () => BedrockRerankingConfiguration$]
];
var RerankQuery$ = [3, n0, _RQ,
    8,
    [_ty, _tQ],
    [0, [() => RerankTextDocument$, 0]]
];
var RerankRequest$ = [3, n0, _RRer,
    0,
    [_q, _so, _rCer, _nTe],
    [[() => RerankQueriesList, 0], [() => RerankSourcesList, 0], () => RerankingConfiguration$, 0]
];
var RerankResponse$ = [3, n0, _RRera,
    0,
    [_res, _nTe],
    [[() => RerankResultsList, 0], 0]
];
var RerankResult$ = [3, n0, _RReran,
    0,
    [_ind, _rSel, _do],
    [1, 1, [() => RerankDocument$, 0]]
];
var RerankSource$ = [3, n0, _RSe,
    8,
    [_ty, _iDS],
    [0, [() => RerankDocument$, 0]]
];
var RerankTextDocument$ = [3, n0, _RTD,
    8,
    [_t],
    [0]
];
var ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ResourceNotFoundException$, ResourceNotFoundException);
var RetrievalResultConfluenceLocation$ = [3, n0, _RRCL,
    0,
    [_ur],
    [0]
];
var RetrievalResultContent$ = [3, n0, _RRC,
    8,
    [_ty, _t, _bC, _vi, _au, _row],
    [0, 0, 0, () => VideoSegment$, () => AudioSegment$, [() => RetrievalResultContentRow, 0]]
];
var RetrievalResultContentColumn$ = [3, n0, _RRCC,
    8,
    [_cNol, _cV, _ty],
    [0, 0, 0]
];
var RetrievalResultCustomDocumentLocation$ = [3, n0, _RRCDL,
    0,
    [_id_],
    [0]
];
var RetrievalResultKendraDocumentLocation$ = [3, n0, _RRKDL,
    0,
    [_uri],
    [0]
];
var RetrievalResultLocation$ = [3, n0, _RRL,
    8,
    [_ty, _sL, _wL, _cL, _sLa, _sPL, _cDL, _kDL, _sLq],
    [0, () => RetrievalResultS3Location$, () => RetrievalResultWebLocation$, () => RetrievalResultConfluenceLocation$, () => RetrievalResultSalesforceLocation$, () => RetrievalResultSharePointLocation$, () => RetrievalResultCustomDocumentLocation$, () => RetrievalResultKendraDocumentLocation$, () => RetrievalResultSqlLocation$]
];
var RetrievalResultS3Location$ = [3, n0, _RRSL,
    0,
    [_uri],
    [0]
];
var RetrievalResultSalesforceLocation$ = [3, n0, _RRSLe,
    0,
    [_ur],
    [0]
];
var RetrievalResultSharePointLocation$ = [3, n0, _RRSPL,
    0,
    [_ur],
    [0]
];
var RetrievalResultSqlLocation$ = [3, n0, _RRSLet,
    0,
    [_qu],
    [0]
];
var RetrievalResultWebLocation$ = [3, n0, _RRWL,
    0,
    [_ur],
    [0]
];
var RetrieveAndGenerateConfiguration$ = [3, n0, _RAGC,
    0,
    [_ty, _kBC, _eSC],
    [0, [() => KnowledgeBaseRetrieveAndGenerateConfiguration$, 0], [() => ExternalSourcesRetrieveAndGenerateConfiguration$, 0]]
];
var RetrieveAndGenerateInput$ = [3, n0, _RAGI,
    8,
    [_t],
    [0]
];
var RetrieveAndGenerateOutput$ = [3, n0, _RAGO,
    8,
    [_t],
    [0]
];
var RetrieveAndGenerateOutputEvent$ = [3, n0, _RAGOE,
    8,
    [_t],
    [0]
];
var RetrieveAndGenerateRequest$ = [3, n0, _RAGR,
    0,
    [_sIe, _i, _rAGC, _sCe],
    [0, [() => RetrieveAndGenerateInput$, 0], [() => RetrieveAndGenerateConfiguration$, 0], () => RetrieveAndGenerateSessionConfiguration$]
];
var RetrieveAndGenerateResponse$ = [3, n0, _RAGRe,
    0,
    [_sIe, _o, _ci, _gA],
    [0, [() => RetrieveAndGenerateOutput$, 0], [() => Citations, 0], 0]
];
var RetrieveAndGenerateSessionConfiguration$ = [3, n0, _RAGSC,
    0,
    [_kKA],
    [0]
];
var RetrieveAndGenerateStreamRequest$ = [3, n0, _RAGSR,
    0,
    [_sIe, _i, _rAGC, _sCe],
    [0, [() => RetrieveAndGenerateInput$, 0], [() => RetrieveAndGenerateConfiguration$, 0], () => RetrieveAndGenerateSessionConfiguration$]
];
var RetrieveAndGenerateStreamResponse$ = [3, n0, _RAGSRe,
    0,
    [_str, _sIe],
    [[() => RetrieveAndGenerateStreamResponseOutput$, 16], [0, { [_hH]: _xabkbsi }]]
];
var RetrievedReference$ = [3, n0, _RRet,
    0,
    [_co, _l, _me],
    [[() => RetrievalResultContent$, 0], [() => RetrievalResultLocation$, 0], [() => RetrievalResultMetadata, 0]]
];
var RetrieveRequest$ = [3, n0, _RRetr,
    0,
    [_kBI, _rQ, _rCe, _gC, _nTe],
    [[0, 1], [() => KnowledgeBaseQuery$, 0], [() => KnowledgeBaseRetrievalConfiguration$, 0], () => GuardrailConfiguration$, 0]
];
var RetrieveResponse$ = [3, n0, _RRetri,
    0,
    [_rRet, _gA, _nTe],
    [[() => KnowledgeBaseRetrievalResults, 0], 0, 0]
];
var ReturnControlPayload$ = [3, n0, _RCP,
    8,
    [_iInv, _iI],
    [[() => InvocationInputs, 0], 0]
];
var ReturnControlResults$ = [3, n0, _RCR,
    0,
    [_iI, _rCIR],
    [0, [() => ReturnControlInvocationResults, 0]]
];
var RoutingClassifierModelInvocationOutput$ = [3, n0, _RCMIO,
    8,
    [_tI, _rRa, _me],
    [0, [() => RawResponse$, 0], [() => Metadata$, 0]]
];
var S3Identifier$ = [3, n0, _SI,
    0,
    [_sBN, _sOK],
    [0, 0]
];
var S3Location$ = [3, n0, _SL,
    0,
    [_uri],
    [0]
];
var S3ObjectDoc$ = [3, n0, _SOD,
    0,
    [_uri],
    [0]
];
var S3ObjectFile$ = [3, n0, _SOF,
    0,
    [_uri],
    [0]
];
var SatisfiedCondition$ = [3, n0, _SC,
    8,
    [_cNo],
    [0]
];
var ServiceQuotaExceededException$ = [-3, n0, _SQEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ServiceQuotaExceededException$, ServiceQuotaExceededException);
var SessionState$ = [3, n0, _SS,
    0,
    [_sAe, _pSA, _rCIR, _iI, _fi, _kBCn, _cH],
    [128 | 0, 128 | 0, [() => ReturnControlInvocationResults, 0], 0, [() => InputFiles, 0], [() => KnowledgeBaseConfigurations, 0], [() => ConversationHistory$, 0]]
];
var SessionSummary$ = [3, n0, _SSe,
    0,
    [_sIe, _sA, _sS, _cA, _lUA],
    [0, 0, 0, 5, 5]
];
var Span$ = [3, n0, _Sp,
    0,
    [_sta, _en],
    [1, 1]
];
var StartFlowExecutionRequest$ = [3, n0, _SFER,
    0,
    [_fI, _fAI, _fEN, _inp, _mPC],
    [[0, 1], [0, 1], 0, [() => FlowInputs, 0], () => ModelPerformanceConfiguration$]
];
var StartFlowExecutionResponse$ = [3, n0, _SFERt,
    0,
    [_eA],
    [0]
];
var StopFlowExecutionRequest$ = [3, n0, _SFERto,
    0,
    [_fI, _fAI, _eI],
    [[0, 1], [0, 1], [0, 1]]
];
var StopFlowExecutionResponse$ = [3, n0, _SFERtop,
    0,
    [_eA, _st],
    [0, 0]
];
var StreamingConfigurations$ = [3, n0, _SCt,
    0,
    [_sFR, _aGI],
    [2, 1]
];
var TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_rA, _ta],
    [[0, 1], 128 | 0]
];
var TagResourceResponse$ = [3, n0, _TRRa,
    0,
    [],
    []
];
var TextInferenceConfig$ = [3, n0, _TIC,
    0,
    [_te, _tPo, _mTa, _sSt],
    [1, 1, 1, 64 | 0]
];
var TextPrompt$ = [3, n0, _TP,
    8,
    [_t],
    [0]
];
var TextResponsePart$ = [3, n0, _TRP,
    8,
    [_t, _sp],
    [0, () => Span$]
];
var TextToSqlConfiguration$ = [3, n0, _TTSC,
    0,
    [_ty, _kBC],
    [0, () => TextToSqlKnowledgeBaseConfiguration$]
];
var TextToSqlKnowledgeBaseConfiguration$ = [3, n0, _TTSKBC,
    0,
    [_kBA],
    [0]
];
var ThrottlingException$ = [-3, n0, _TE,
    { [_e]: _c, [_hE]: 429 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ThrottlingException$, ThrottlingException);
var TracePart$ = [3, n0, _TPr,
    8,
    [_sIe, _tra, _cCa, _eTv, _cN, _aI, _aAI, _aV],
    [0, [() => Trace$, 0], () => CallerChain, 5, [() => Name, 0], 0, 0, 0]
];
var TransformationConfiguration$ = [3, n0, _TC,
    0,
    [_mo, _tTSC],
    [0, () => TextToSqlConfiguration$]
];
var UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_rA, _tKa],
    [[0, 1], [64 | 0, { [_hQ]: _tKa }]]
];
var UntagResourceResponse$ = [3, n0, _URRn,
    0,
    [],
    []
];
var UpdateSessionRequest$ = [3, n0, _USR,
    0,
    [_sM, _sI],
    [128 | 0, [0, 1]]
];
var UpdateSessionResponse$ = [3, n0, _USRp,
    0,
    [_sIe, _sA, _sS, _cA, _lUA],
    [0, 0, 0, 5, 5]
];
var Usage$ = [3, n0, _U,
    8,
    [_iTnp, _oTu],
    [1, 1]
];
var ValidationException$ = [-3, n0, _VE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ValidationException$, ValidationException);
var VectorSearchBedrockRerankingConfiguration$ = [3, n0, _VSBRC,
    0,
    [_mC, _nORR, _mCet],
    [() => VectorSearchBedrockRerankingModelConfiguration$, 1, [() => MetadataConfigurationForReranking$, 0]]
];
var VectorSearchBedrockRerankingModelConfiguration$ = [3, n0, _VSBRMC,
    0,
    [_mA, _aMRF],
    [0, 128 | 15]
];
var VectorSearchRerankingConfiguration$ = [3, n0, _VSRC,
    0,
    [_ty, _bRC],
    [0, [() => VectorSearchBedrockRerankingConfiguration$, 0]]
];
var VideoSegment$ = [3, n0, _VS,
    0,
    [_sU, _su],
    [0, 0]
];
var BedrockAgentRuntimeServiceException$ = [-3, _sm, "BedrockAgentRuntimeServiceException", 0, [], []];
schema.TypeRegistry.for(_sm).registerError(BedrockAgentRuntimeServiceException$, BedrockAgentRuntimeServiceException);
var AgentActionGroups = [1, n0, _AAGg,
    0, [() => AgentActionGroup$,
        0]
];
var AgentTraces = [1, n0, _AT,
    0, [() => TracePart$,
        0]
];
var ApiParameters = [1, n0, _APpi,
    0, () => ApiParameter$
];
var BedrockSessionContentBlocks = [1, n0, _BSCB,
    0, [() => BedrockSessionContentBlock$,
        0]
];
var CallerChain = [1, n0, _CCa,
    0, () => Caller$
];
var Citations = [1, n0, _Ci,
    0, [() => Citation$,
        0]
];
var CollaboratorConfigurations = [1, n0, _CCo,
    0, [() => CollaboratorConfiguration$,
        0]
];
var Collaborators = [1, n0, _Col,
    0, [() => Collaborator$,
        0]
];
var ContentBlocks = [1, n0, _CBo,
    0, [() => ContentBlock$,
        0]
];
var ExternalSources = [1, n0, _ESx,
    0, [() => ExternalSource$,
        0]
];
var FieldsForReranking = [1, n0, _FFRi,
    8, () => FieldForReranking$
];
var FlowExecutionErrors = [1, n0, _FEEl,
    0, () => FlowExecutionError$
];
var FlowExecutionEvents = [1, n0, _FEElo,
    0, [() => FlowExecutionEvent$,
        0]
];
var FlowExecutionSummaries = [1, n0, _FESl,
    0, () => FlowExecutionSummary$
];
var FlowInputFields = [1, n0, _FIFl,
    0, [() => FlowInputField$,
        0]
];
var FlowInputs = [1, n0, _FIl,
    0, [() => FlowInput$,
        0]
];
var FlowOutputFields = [1, n0, _FOFl,
    0, [() => FlowOutputField$,
        0]
];
var FlowTraceConditions = [1, n0, _FTCl,
    0, [() => FlowTraceCondition$,
        0]
];
var FlowTraceNodeInputExecutionChain = [1, n0, _FTNIEC,
    8, [() => FlowTraceNodeInputExecutionChainItem$,
        0]
];
var FlowTraceNodeInputFields = [1, n0, _FTNIFl,
    0, [() => FlowTraceNodeInputField$,
        0]
];
var FlowTraceNodeOutputFields = [1, n0, _FTNOFl,
    0, [() => FlowTraceNodeOutputField$,
        0]
];
var FlowTraceNodeOutputNextList = [1, n0, _FTNONL,
    0, [() => FlowTraceNodeOutputNext$,
        0]
];
var FunctionParameters = [1, n0, _FPun,
    0, () => FunctionParameter$
];
var Functions = [1, n0, _Fu,
    0, [() => FunctionDefinition$,
        0]
];
var GeneratedQueries = [1, n0, _GQe,
    0, [() => GeneratedQuery$,
        0]
];
var GuardrailAssessmentList = [1, n0, _GAL,
    0, [() => GuardrailAssessment$,
        0]
];
var GuardrailContentFilterList = [1, n0, _GCFL,
    8, [() => GuardrailContentFilter$,
        0]
];
var GuardrailCustomWordList = [1, n0, _GCWL,
    8, [() => GuardrailCustomWord$,
        0]
];
var GuardrailManagedWordList = [1, n0, _GMWL,
    8, [() => GuardrailManagedWord$,
        0]
];
var GuardrailPiiEntityFilterList = [1, n0, _GPEFL,
    8, [() => GuardrailPiiEntityFilter$,
        0]
];
var GuardrailRegexFilterList = [1, n0, _GRFL,
    8, [() => GuardrailRegexFilter$,
        0]
];
var GuardrailTopicList = [1, n0, _GTL,
    8, [() => GuardrailTopic$,
        0]
];
var ImageInputs = [1, n0, _IIm,
    0, () => ImageInput$
];
var InputFiles = [1, n0, _IFn,
    0, [() => InputFile$,
        0]
];
var InvocationInputs = [1, n0, _IInvo,
    0, [() => InvocationInputMember$,
        0]
];
var InvocationStepSummaries = [1, n0, _ISSnv,
    0, () => InvocationStepSummary$
];
var InvocationSummaries = [1, n0, _ISnv,
    0, () => InvocationSummary$
];
var KnowledgeBaseConfigurations = [1, n0, _KBCn,
    0, [() => KnowledgeBaseConfiguration$,
        0]
];
var KnowledgeBaseRetrievalResults = [1, n0, _KBRRn,
    8, [() => KnowledgeBaseRetrievalResult$,
        0]
];
var KnowledgeBases = [1, n0, _KBn,
    0, [() => KnowledgeBase$,
        0]
];
var Memories = [1, n0, _Mem,
    0, () => Memory$
];
var Messages = [1, n0, _Mes,
    0, [() => Message$,
        0]
];
var MetadataAttributeSchemaList = [1, n0, _MASL,
    0, [() => MetadataAttributeSchema$,
        0]
];
var NodeInputExecutionChain = [1, n0, _NIEC,
    0, () => NodeInputExecutionChainItem$
];
var NodeInputFields = [1, n0, _NIFo,
    0, [() => NodeInputField$,
        0]
];
var NodeOutputFields = [1, n0, _NOFo,
    0, [() => NodeOutputField$,
        0]
];
var NodeOutputNextList = [1, n0, _NONL,
    0, [() => NodeOutputNext$,
        0]
];
var OutputFiles = [1, n0, _OFu,
    0, [() => OutputFile$,
        0]
];
var ParameterList = [1, n0, _PL,
    0, () => Parameter$
];
var _Parameters = [1, n0, _Par,
    0, () => Parameter$
];
var PromptConfigurations = [1, n0, _PCro,
    0, [() => PromptConfiguration$,
        0]
];
var RerankQueriesList = [1, n0, _RQL,
    8, [() => RerankQuery$,
        0]
];
var RerankResultsList = [1, n0, _RRLe,
    0, [() => RerankResult$,
        0]
];
var RerankSourcesList = [1, n0, _RSL,
    8, [() => RerankSource$,
        0]
];
var RetrievalFilterList = [1, n0, _RFL,
    0, [() => RetrievalFilter$,
        0]
];
var RetrievalResultContentRow = [1, n0, _RRCR,
    8, [() => RetrievalResultContentColumn$,
        0]
];
var RetrievedReferences = [1, n0, _RRetrie,
    0, [() => RetrievedReference$,
        0]
];
var ReturnControlInvocationResults = [1, n0, _RCIR,
    0, [() => InvocationResultMember$,
        0]
];
var SatisfiedConditions = [1, n0, _SCa,
    0, [() => SatisfiedCondition$,
        0]
];
var SessionSummaries = [1, n0, _SSes,
    0, () => SessionSummary$
];
var ApiContentMap = [2, n0, _ACM,
    0, 0, () => PropertyParameters$
];
var ContentMap = [2, n0, _CM,
    0, 0, () => _Parameters
];
var ParameterMap = [2, n0, _PM,
    0, 0, () => ParameterDetail$
];
var ResponseBody = [2, n0, _RBe,
    0, 0, () => ContentBody$
];
var RetrievalResultMetadata = [2, n0, _RRM,
    8, 0, 15
];
var ActionGroupExecutor$ = [4, n0, _AGE,
    0,
    [_lam, _cCu],
    [0, 0]
];
var APISchema$ = [4, n0, _APIS,
    0,
    [_s_, _pa],
    [() => S3Identifier$, [() => Payload, 0]]
];
var BedrockSessionContentBlock$ = [4, n0, _BSCBe,
    8,
    [_t, _ima],
    [0, () => ImageBlock$]
];
var Caller$ = [4, n0, _Ca,
    0,
    [_aAA],
    [0]
];
var ContentBlock$ = [4, n0, _CBon,
    8,
    [_t],
    [0]
];
var FlowExecutionContent$ = [4, n0, _FEC,
    8,
    [_do],
    [15]
];
var FlowExecutionEvent$ = [4, n0, _FEElow,
    0,
    [_fIE, _fOE, _nIE, _nOE, _cRE, _nFE, _fFE, _nAE, _nDE],
    [[() => FlowExecutionInputEvent$, 0], [() => FlowExecutionOutputEvent$, 0], [() => NodeInputEvent$, 0], [() => NodeOutputEvent$, 0], [() => ConditionResultEvent$, 0], [() => NodeFailureEvent$, 0], [() => FlowFailureEvent$, 0], [() => NodeActionEvent$, 0], [() => NodeDependencyEvent$, 0]]
];
var FlowInputContent$ = [4, n0, _FIC,
    8,
    [_do],
    [15]
];
var FlowMultiTurnInputContent$ = [4, n0, _FMTIC,
    0,
    [_do],
    [15]
];
var FlowOutputContent$ = [4, n0, _FOC,
    0,
    [_do],
    [15]
];
var FlowResponseStream$ = [4, n0, _FRSl,
    { [_stre]: 1 },
    [_fOE, _fCE, _fTE, _iSE, _vE, _rNFE, _sQEE, _tEh, _aDE, _cE, _dFE, _bGE, _fMTIRE],
    [[() => FlowOutputEvent$, 0], [() => FlowCompletionEvent$, 0], [() => FlowTraceEvent$, 0], [() => InternalServerException$, 0], [() => ValidationException$, 0], [() => ResourceNotFoundException$, 0], [() => ServiceQuotaExceededException$, 0], [() => ThrottlingException$, 0], [() => AccessDeniedException$, 0], [() => ConflictException$, 0], [() => DependencyFailedException$, 0], [() => BadGatewayException$, 0], [() => FlowMultiTurnInputRequestEvent$, 0]]
];
var FlowTrace$ = [4, n0, _FTl,
    8,
    [_nIT, _nOT, _cNRT, _nAT, _nDT],
    [[() => FlowTraceNodeInputEvent$, 0], [() => FlowTraceNodeOutputEvent$, 0], [() => FlowTraceConditionNodeResultEvent$, 0], [() => FlowTraceNodeActionEvent$, 0], [() => FlowTraceDependencyEvent$, 0]]
];
var FlowTraceNodeInputContent$ = [4, n0, _FTNIC,
    8,
    [_do],
    [15]
];
var FlowTraceNodeOutputContent$ = [4, n0, _FTNOC,
    0,
    [_do],
    [15]
];
var FunctionSchema$ = [4, n0, _FSu,
    0,
    [_fu],
    [[() => Functions, 0]]
];
var ImageInputSource$ = [4, n0, _IIS,
    0,
    [_by],
    [21]
];
var ImageSource$ = [4, n0, _ISm,
    0,
    [_by, _sL],
    [21, () => S3Location$]
];
var InlineAgentResponseStream$ = [4, n0, _IARS,
    { [_stre]: 1 },
    [_ch, _tra, _rCet, _iSE, _vE, _rNFE, _sQEE, _tEh, _aDE, _cE, _dFE, _bGE, _fi],
    [[() => InlineAgentPayloadPart$, 0], [() => InlineAgentTracePart$, 0], [() => InlineAgentReturnControlPayload$, 0], [() => InternalServerException$, 0], [() => ValidationException$, 0], [() => ResourceNotFoundException$, 0], [() => ServiceQuotaExceededException$, 0], [() => ThrottlingException$, 0], [() => AccessDeniedException$, 0], [() => ConflictException$, 0], [() => DependencyFailedException$, 0], [() => BadGatewayException$, 0], [() => InlineAgentFilePart$, 0]]
];
var InputPrompt$ = [4, n0, _IP,
    0,
    [_tPe],
    [[() => TextPrompt$, 0]]
];
var InvocationInputMember$ = [4, n0, _IIM,
    0,
    [_aII, _fII],
    [[() => ApiInvocationInput$, 0], [() => FunctionInvocationInput$, 0]]
];
var InvocationResultMember$ = [4, n0, _IRM,
    0,
    [_aR, _fRu],
    [[() => ApiResult$, 0], () => FunctionResult$]
];
var InvocationStepPayload$ = [4, n0, _ISP,
    0,
    [_cB],
    [[() => BedrockSessionContentBlocks, 0]]
];
var Memory$ = [4, n0, _Memo,
    0,
    [_sSess],
    [() => MemorySessionSummary$]
];
var NodeExecutionContent$ = [4, n0, _NEC,
    8,
    [_do],
    [15]
];
var NodeTraceElements$ = [4, n0, _NTE,
    8,
    [_aT],
    [[() => AgentTraces, 0]]
];
var OptimizedPrompt$ = [4, n0, _OP,
    0,
    [_tPe],
    [[() => TextPrompt$, 0]]
];
var OptimizedPromptStream$ = [4, n0, _OPS,
    { [_stre]: 1 },
    [_oPE, _aPE, _iSE, _tEh, _vE, _dFE, _aDE, _bGE],
    [[() => OptimizedPromptEvent$, 0], [() => AnalyzePromptEvent$, 0], [() => InternalServerException$, 0], [() => ThrottlingException$, 0], [() => ValidationException$, 0], [() => DependencyFailedException$, 0], [() => AccessDeniedException$, 0], [() => BadGatewayException$, 0]]
];
var OrchestrationExecutor$ = [4, n0, _OE,
    0,
    [_lam],
    [0]
];
var OrchestrationTrace$ = [4, n0, _OT,
    8,
    [_ra, _iInvo, _ob, _mII, _mIO],
    [[() => Rationale$, 0], [() => InvocationInput$, 0], [() => Observation$, 0], [() => ModelInvocationInput$, 0], [() => OrchestrationModelInvocationOutput$, 0]]
];
var PostProcessingTrace$ = [4, n0, _PPT,
    8,
    [_mII, _mIO],
    [[() => ModelInvocationInput$, 0], [() => PostProcessingModelInvocationOutput$, 0]]
];
var PreProcessingTrace$ = [4, n0, _PPTr,
    8,
    [_mII, _mIO],
    [[() => ModelInvocationInput$, 0], [() => PreProcessingModelInvocationOutput$, 0]]
];
var ReasoningContentBlock$ = [4, n0, _RCB,
    8,
    [_rT, _rCed],
    [[() => ReasoningTextBlock$, 0], 21]
];
var RerankingMetadataSelectiveModeConfiguration$ = [4, n0, _RMSMC,
    0,
    [_fTI, _fTEi],
    [[() => FieldsForReranking, 0], [() => FieldsForReranking, 0]]
];
var ResponseStream$ = [4, n0, _RSes,
    { [_stre]: 1 },
    [_ch, _tra, _rCet, _iSE, _vE, _rNFE, _sQEE, _tEh, _aDE, _cE, _dFE, _bGE, _mNRE, _fi],
    [[() => PayloadPart$, 0], [() => TracePart$, 0], [() => ReturnControlPayload$, 0], [() => InternalServerException$, 0], [() => ValidationException$, 0], [() => ResourceNotFoundException$, 0], [() => ServiceQuotaExceededException$, 0], [() => ThrottlingException$, 0], [() => AccessDeniedException$, 0], [() => ConflictException$, 0], [() => DependencyFailedException$, 0], [() => BadGatewayException$, 0], [() => ModelNotReadyException$, 0], [() => FilePart$, 0]]
];
var RetrievalFilter$ = [4, n0, _RF,
    8,
    [_eq, _nE, _gT, _gTOE, _lT, _lTOE, _in_, _nI, _sW, _lC, _sCtr, _aA, _oAr],
    [() => FilterAttribute$, () => FilterAttribute$, () => FilterAttribute$, () => FilterAttribute$, () => FilterAttribute$, () => FilterAttribute$, () => FilterAttribute$, () => FilterAttribute$, () => FilterAttribute$, () => FilterAttribute$, () => FilterAttribute$, [() => RetrievalFilterList, 0], [() => RetrievalFilterList, 0]]
];
var RetrieveAndGenerateStreamResponseOutput$ = [4, n0, _RAGSRO,
    { [_stre]: 1 },
    [_o, _cit, _g, _iSE, _vE, _rNFE, _sQEE, _tEh, _aDE, _cE, _dFE, _bGE],
    [[() => RetrieveAndGenerateOutputEvent$, 0], [() => CitationEvent$, 0], () => GuardrailEvent$, [() => InternalServerException$, 0], [() => ValidationException$, 0], [() => ResourceNotFoundException$, 0], [() => ServiceQuotaExceededException$, 0], [() => ThrottlingException$, 0], [() => AccessDeniedException$, 0], [() => ConflictException$, 0], [() => DependencyFailedException$, 0], [() => BadGatewayException$, 0]]
];
var RoutingClassifierTrace$ = [4, n0, _RCT,
    8,
    [_iInvo, _ob, _mII, _mIO],
    [[() => InvocationInput$, 0], [() => Observation$, 0], [() => ModelInvocationInput$, 0], [() => RoutingClassifierModelInvocationOutput$, 0]]
];
var Trace$ = [4, n0, _T,
    8,
    [_gTu, _pPT, _oTr, _pPTo, _rCT, _fT, _cOT],
    [[() => GuardrailTrace$, 0], [() => PreProcessingTrace$, 0], [() => OrchestrationTrace$, 0], [() => PostProcessingTrace$, 0], [() => RoutingClassifierTrace$, 0], [() => FailureTrace$, 0], [() => CustomOrchestrationTrace$, 0]]
];
var TraceElements$ = [4, n0, _TEr,
    8,
    [_aT],
    [[() => AgentTraces, 0]]
];
var CreateInvocation$ = [9, n0, _CIr,
    { [_h]: ["PUT", "/sessions/{sessionIdentifier}/invocations/", 201] }, () => CreateInvocationRequest$, () => CreateInvocationResponse$
];
var CreateSession$ = [9, n0, _CS,
    { [_h]: ["PUT", "/sessions/", 201] }, () => CreateSessionRequest$, () => CreateSessionResponse$
];
var DeleteAgentMemory$ = [9, n0, _DAM,
    { [_h]: ["DELETE", "/agents/{agentId}/agentAliases/{agentAliasId}/memories", 202] }, () => DeleteAgentMemoryRequest$, () => DeleteAgentMemoryResponse$
];
var DeleteSession$ = [9, n0, _DS,
    { [_h]: ["DELETE", "/sessions/{sessionIdentifier}/", 200] }, () => DeleteSessionRequest$, () => DeleteSessionResponse$
];
var EndSession$ = [9, n0, _ESn,
    { [_h]: ["PATCH", "/sessions/{sessionIdentifier}", 200] }, () => EndSessionRequest$, () => EndSessionResponse$
];
var GenerateQuery$ = [9, n0, _GQen,
    { [_h]: ["POST", "/generateQuery", 200] }, () => GenerateQueryRequest$, () => GenerateQueryResponse$
];
var GetAgentMemory$ = [9, n0, _GAM,
    { [_h]: ["GET", "/agents/{agentId}/agentAliases/{agentAliasId}/memories", 200] }, () => GetAgentMemoryRequest$, () => GetAgentMemoryResponse$
];
var GetExecutionFlowSnapshot$ = [9, n0, _GEFS,
    { [_h]: ["GET", "/flows/{flowIdentifier}/aliases/{flowAliasIdentifier}/executions/{executionIdentifier}/flowsnapshot", 200] }, () => GetExecutionFlowSnapshotRequest$, () => GetExecutionFlowSnapshotResponse$
];
var GetFlowExecution$ = [9, n0, _GFE,
    { [_h]: ["GET", "/flows/{flowIdentifier}/aliases/{flowAliasIdentifier}/executions/{executionIdentifier}", 200] }, () => GetFlowExecutionRequest$, () => GetFlowExecutionResponse$
];
var GetInvocationStep$ = [9, n0, _GIS,
    { [_h]: ["POST", "/sessions/{sessionIdentifier}/invocationSteps/{invocationStepId}", 200] }, () => GetInvocationStepRequest$, () => GetInvocationStepResponse$
];
var GetSession$ = [9, n0, _GS,
    { [_h]: ["GET", "/sessions/{sessionIdentifier}/", 200] }, () => GetSessionRequest$, () => GetSessionResponse$
];
var InvokeAgent$ = [9, n0, _IA,
    { [_h]: ["POST", "/agents/{agentId}/agentAliases/{agentAliasId}/sessions/{sessionId}/text", 200] }, () => InvokeAgentRequest$, () => InvokeAgentResponse$
];
var InvokeFlow$ = [9, n0, _IFnv,
    { [_h]: ["POST", "/flows/{flowIdentifier}/aliases/{flowAliasIdentifier}", 200] }, () => InvokeFlowRequest$, () => InvokeFlowResponse$
];
var InvokeInlineAgent$ = [9, n0, _IIA,
    { [_h]: ["POST", "/agents/{sessionId}", 200] }, () => InvokeInlineAgentRequest$, () => InvokeInlineAgentResponse$
];
var ListFlowExecutionEvents$ = [9, n0, _LFEE,
    { [_h]: ["GET", "/flows/{flowIdentifier}/aliases/{flowAliasIdentifier}/executions/{executionIdentifier}/events", 200] }, () => ListFlowExecutionEventsRequest$, () => ListFlowExecutionEventsResponse$
];
var ListFlowExecutions$ = [9, n0, _LFE,
    { [_h]: ["GET", "/flows/{flowIdentifier}/executions", 200] }, () => ListFlowExecutionsRequest$, () => ListFlowExecutionsResponse$
];
var ListInvocations$ = [9, n0, _LI,
    { [_h]: ["POST", "/sessions/{sessionIdentifier}/invocations/", 200] }, () => ListInvocationsRequest$, () => ListInvocationsResponse$
];
var ListInvocationSteps$ = [9, n0, _LIS,
    { [_h]: ["POST", "/sessions/{sessionIdentifier}/invocationSteps/", 200] }, () => ListInvocationStepsRequest$, () => ListInvocationStepsResponse$
];
var ListSessions$ = [9, n0, _LS,
    { [_h]: ["POST", "/sessions/", 200] }, () => ListSessionsRequest$, () => ListSessionsResponse$
];
var ListTagsForResource$ = [9, n0, _LTFR,
    { [_h]: ["GET", "/tags/{resourceArn}", 200] }, () => ListTagsForResourceRequest$, () => ListTagsForResourceResponse$
];
var OptimizePrompt$ = [9, n0, _OPp,
    { [_h]: ["POST", "/optimize-prompt", 200] }, () => OptimizePromptRequest$, () => OptimizePromptResponse$
];
var PutInvocationStep$ = [9, n0, _PIS,
    { [_h]: ["PUT", "/sessions/{sessionIdentifier}/invocationSteps/", 201] }, () => PutInvocationStepRequest$, () => PutInvocationStepResponse$
];
var Rerank$ = [9, n0, _Re,
    { [_h]: ["POST", "/rerank", 200] }, () => RerankRequest$, () => RerankResponse$
];
var Retrieve$ = [9, n0, _Ret,
    { [_h]: ["POST", "/knowledgebases/{knowledgeBaseId}/retrieve", 200] }, () => RetrieveRequest$, () => RetrieveResponse$
];
var RetrieveAndGenerate$ = [9, n0, _RAG,
    { [_h]: ["POST", "/retrieveAndGenerate", 200] }, () => RetrieveAndGenerateRequest$, () => RetrieveAndGenerateResponse$
];
var RetrieveAndGenerateStream$ = [9, n0, _RAGS,
    { [_h]: ["POST", "/retrieveAndGenerateStream", 200] }, () => RetrieveAndGenerateStreamRequest$, () => RetrieveAndGenerateStreamResponse$
];
var StartFlowExecution$ = [9, n0, _SFE,
    { [_h]: ["POST", "/flows/{flowIdentifier}/aliases/{flowAliasIdentifier}/executions", 200] }, () => StartFlowExecutionRequest$, () => StartFlowExecutionResponse$
];
var StopFlowExecution$ = [9, n0, _SFEt,
    { [_h]: ["POST", "/flows/{flowIdentifier}/aliases/{flowAliasIdentifier}/executions/{executionIdentifier}/stop", 200] }, () => StopFlowExecutionRequest$, () => StopFlowExecutionResponse$
];
var TagResource$ = [9, n0, _TR,
    { [_h]: ["POST", "/tags/{resourceArn}", 200] }, () => TagResourceRequest$, () => TagResourceResponse$
];
var UntagResource$ = [9, n0, _UR,
    { [_h]: ["DELETE", "/tags/{resourceArn}", 200] }, () => UntagResourceRequest$, () => UntagResourceResponse$
];
var UpdateSession$ = [9, n0, _US,
    { [_h]: ["PUT", "/sessions/{sessionIdentifier}/", 200] }, () => UpdateSessionRequest$, () => UpdateSessionResponse$
];

class CreateInvocationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "CreateInvocation", {})
    .n("BedrockAgentRuntimeClient", "CreateInvocationCommand")
    .sc(CreateInvocation$)
    .build() {
}

class CreateSessionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "CreateSession", {})
    .n("BedrockAgentRuntimeClient", "CreateSessionCommand")
    .sc(CreateSession$)
    .build() {
}

class DeleteAgentMemoryCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "DeleteAgentMemory", {})
    .n("BedrockAgentRuntimeClient", "DeleteAgentMemoryCommand")
    .sc(DeleteAgentMemory$)
    .build() {
}

class DeleteSessionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "DeleteSession", {})
    .n("BedrockAgentRuntimeClient", "DeleteSessionCommand")
    .sc(DeleteSession$)
    .build() {
}

class EndSessionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "EndSession", {})
    .n("BedrockAgentRuntimeClient", "EndSessionCommand")
    .sc(EndSession$)
    .build() {
}

class GenerateQueryCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "GenerateQuery", {})
    .n("BedrockAgentRuntimeClient", "GenerateQueryCommand")
    .sc(GenerateQuery$)
    .build() {
}

class GetAgentMemoryCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "GetAgentMemory", {})
    .n("BedrockAgentRuntimeClient", "GetAgentMemoryCommand")
    .sc(GetAgentMemory$)
    .build() {
}

class GetExecutionFlowSnapshotCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "GetExecutionFlowSnapshot", {})
    .n("BedrockAgentRuntimeClient", "GetExecutionFlowSnapshotCommand")
    .sc(GetExecutionFlowSnapshot$)
    .build() {
}

class GetFlowExecutionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "GetFlowExecution", {})
    .n("BedrockAgentRuntimeClient", "GetFlowExecutionCommand")
    .sc(GetFlowExecution$)
    .build() {
}

class GetInvocationStepCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "GetInvocationStep", {})
    .n("BedrockAgentRuntimeClient", "GetInvocationStepCommand")
    .sc(GetInvocationStep$)
    .build() {
}

class GetSessionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "GetSession", {})
    .n("BedrockAgentRuntimeClient", "GetSessionCommand")
    .sc(GetSession$)
    .build() {
}

class InvokeAgentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "InvokeAgent", {
    eventStream: {
        output: true,
    },
})
    .n("BedrockAgentRuntimeClient", "InvokeAgentCommand")
    .sc(InvokeAgent$)
    .build() {
}

class InvokeFlowCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "InvokeFlow", {
    eventStream: {
        output: true,
    },
})
    .n("BedrockAgentRuntimeClient", "InvokeFlowCommand")
    .sc(InvokeFlow$)
    .build() {
}

class InvokeInlineAgentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "InvokeInlineAgent", {
    eventStream: {
        output: true,
    },
})
    .n("BedrockAgentRuntimeClient", "InvokeInlineAgentCommand")
    .sc(InvokeInlineAgent$)
    .build() {
}

class ListFlowExecutionEventsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "ListFlowExecutionEvents", {})
    .n("BedrockAgentRuntimeClient", "ListFlowExecutionEventsCommand")
    .sc(ListFlowExecutionEvents$)
    .build() {
}

class ListFlowExecutionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "ListFlowExecutions", {})
    .n("BedrockAgentRuntimeClient", "ListFlowExecutionsCommand")
    .sc(ListFlowExecutions$)
    .build() {
}

class ListInvocationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "ListInvocations", {})
    .n("BedrockAgentRuntimeClient", "ListInvocationsCommand")
    .sc(ListInvocations$)
    .build() {
}

class ListInvocationStepsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "ListInvocationSteps", {})
    .n("BedrockAgentRuntimeClient", "ListInvocationStepsCommand")
    .sc(ListInvocationSteps$)
    .build() {
}

class ListSessionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "ListSessions", {})
    .n("BedrockAgentRuntimeClient", "ListSessionsCommand")
    .sc(ListSessions$)
    .build() {
}

class ListTagsForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "ListTagsForResource", {})
    .n("BedrockAgentRuntimeClient", "ListTagsForResourceCommand")
    .sc(ListTagsForResource$)
    .build() {
}

class OptimizePromptCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "OptimizePrompt", {
    eventStream: {
        output: true,
    },
})
    .n("BedrockAgentRuntimeClient", "OptimizePromptCommand")
    .sc(OptimizePrompt$)
    .build() {
}

class PutInvocationStepCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "PutInvocationStep", {})
    .n("BedrockAgentRuntimeClient", "PutInvocationStepCommand")
    .sc(PutInvocationStep$)
    .build() {
}

class RerankCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "Rerank", {})
    .n("BedrockAgentRuntimeClient", "RerankCommand")
    .sc(Rerank$)
    .build() {
}

class RetrieveAndGenerateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "RetrieveAndGenerate", {})
    .n("BedrockAgentRuntimeClient", "RetrieveAndGenerateCommand")
    .sc(RetrieveAndGenerate$)
    .build() {
}

class RetrieveAndGenerateStreamCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "RetrieveAndGenerateStream", {
    eventStream: {
        output: true,
    },
})
    .n("BedrockAgentRuntimeClient", "RetrieveAndGenerateStreamCommand")
    .sc(RetrieveAndGenerateStream$)
    .build() {
}

class RetrieveCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "Retrieve", {})
    .n("BedrockAgentRuntimeClient", "RetrieveCommand")
    .sc(Retrieve$)
    .build() {
}

class StartFlowExecutionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "StartFlowExecution", {})
    .n("BedrockAgentRuntimeClient", "StartFlowExecutionCommand")
    .sc(StartFlowExecution$)
    .build() {
}

class StopFlowExecutionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "StopFlowExecution", {})
    .n("BedrockAgentRuntimeClient", "StopFlowExecutionCommand")
    .sc(StopFlowExecution$)
    .build() {
}

class TagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "TagResource", {})
    .n("BedrockAgentRuntimeClient", "TagResourceCommand")
    .sc(TagResource$)
    .build() {
}

class UntagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "UntagResource", {})
    .n("BedrockAgentRuntimeClient", "UntagResourceCommand")
    .sc(UntagResource$)
    .build() {
}

class UpdateSessionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonBedrockAgentRunTimeService", "UpdateSession", {})
    .n("BedrockAgentRuntimeClient", "UpdateSessionCommand")
    .sc(UpdateSession$)
    .build() {
}

const commands = {
    CreateInvocationCommand,
    CreateSessionCommand,
    DeleteAgentMemoryCommand,
    DeleteSessionCommand,
    EndSessionCommand,
    GenerateQueryCommand,
    GetAgentMemoryCommand,
    GetExecutionFlowSnapshotCommand,
    GetFlowExecutionCommand,
    GetInvocationStepCommand,
    GetSessionCommand,
    InvokeAgentCommand,
    InvokeFlowCommand,
    InvokeInlineAgentCommand,
    ListFlowExecutionEventsCommand,
    ListFlowExecutionsCommand,
    ListInvocationsCommand,
    ListInvocationStepsCommand,
    ListSessionsCommand,
    ListTagsForResourceCommand,
    OptimizePromptCommand,
    PutInvocationStepCommand,
    RerankCommand,
    RetrieveCommand,
    RetrieveAndGenerateCommand,
    RetrieveAndGenerateStreamCommand,
    StartFlowExecutionCommand,
    StopFlowExecutionCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateSessionCommand,
};
class BedrockAgentRuntime extends BedrockAgentRuntimeClient {
}
smithyClient.createAggregatedClient(commands, BedrockAgentRuntime);

const paginateGetAgentMemory = core.createPaginator(BedrockAgentRuntimeClient, GetAgentMemoryCommand, "nextToken", "nextToken", "maxItems");

const paginateListFlowExecutionEvents = core.createPaginator(BedrockAgentRuntimeClient, ListFlowExecutionEventsCommand, "nextToken", "nextToken", "maxResults");

const paginateListFlowExecutions = core.createPaginator(BedrockAgentRuntimeClient, ListFlowExecutionsCommand, "nextToken", "nextToken", "maxResults");

const paginateListInvocations = core.createPaginator(BedrockAgentRuntimeClient, ListInvocationsCommand, "nextToken", "nextToken", "maxResults");

const paginateListInvocationSteps = core.createPaginator(BedrockAgentRuntimeClient, ListInvocationStepsCommand, "nextToken", "nextToken", "maxResults");

const paginateListSessions = core.createPaginator(BedrockAgentRuntimeClient, ListSessionsCommand, "nextToken", "nextToken", "maxResults");

const paginateRerank = core.createPaginator(BedrockAgentRuntimeClient, RerankCommand, "nextToken", "nextToken", "");

const paginateRetrieve = core.createPaginator(BedrockAgentRuntimeClient, RetrieveCommand, "nextToken", "nextToken", "");

const CustomControlMethod = {
    RETURN_CONTROL: "RETURN_CONTROL",
};
const ExecutionType = {
    LAMBDA: "LAMBDA",
    RETURN_CONTROL: "RETURN_CONTROL",
};
const ActionGroupSignature = {
    AMAZON_CODEINTERPRETER: "AMAZON.CodeInterpreter",
    AMAZON_USERINPUT: "AMAZON.UserInput",
    ANTHROPIC_BASH: "ANTHROPIC.Bash",
    ANTHROPIC_COMPUTER: "ANTHROPIC.Computer",
    ANTHROPIC_TEXTEDITOR: "ANTHROPIC.TextEditor",
};
const ActionInvocationType = {
    RESULT: "RESULT",
    USER_CONFIRMATION: "USER_CONFIRMATION",
    USER_CONFIRMATION_AND_RESULT: "USER_CONFIRMATION_AND_RESULT",
};
const ParameterType = {
    ARRAY: "array",
    BOOLEAN: "boolean",
    INTEGER: "integer",
    NUMBER: "number",
    STRING: "string",
};
const RequireConfirmation = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const AgentCollaboration = {
    DISABLED: "DISABLED",
    SUPERVISOR: "SUPERVISOR",
    SUPERVISOR_ROUTER: "SUPERVISOR_ROUTER",
};
const ConfirmationState = {
    CONFIRM: "CONFIRM",
    DENY: "DENY",
};
const ImageInputFormat = {
    GIF: "gif",
    JPEG: "jpeg",
    PNG: "png",
    WEBP: "webp",
};
const ResponseState = {
    FAILURE: "FAILURE",
    REPROMPT: "REPROMPT",
};
const PayloadType = {
    RETURN_CONTROL: "RETURN_CONTROL",
    TEXT: "TEXT",
};
const GuardrailAction = {
    INTERVENED: "INTERVENED",
    NONE: "NONE",
};
const GuardrailContentPolicyAction = {
    BLOCKED: "BLOCKED",
};
const GuardrailContentFilterConfidence = {
    HIGH: "HIGH",
    LOW: "LOW",
    MEDIUM: "MEDIUM",
    NONE: "NONE",
};
const GuardrailContentFilterType = {
    HATE: "HATE",
    INSULTS: "INSULTS",
    MISCONDUCT: "MISCONDUCT",
    PROMPT_ATTACK: "PROMPT_ATTACK",
    SEXUAL: "SEXUAL",
    VIOLENCE: "VIOLENCE",
};
const GuardrailSensitiveInformationPolicyAction = {
    ANONYMIZED: "ANONYMIZED",
    BLOCKED: "BLOCKED",
};
const GuardrailPiiEntityType = {
    ADDRESS: "ADDRESS",
    AGE: "AGE",
    AWS_ACCESS_KEY: "AWS_ACCESS_KEY",
    AWS_SECRET_KEY: "AWS_SECRET_KEY",
    CA_HEALTH_NUMBER: "CA_HEALTH_NUMBER",
    CA_SOCIAL_INSURANCE_NUMBER: "CA_SOCIAL_INSURANCE_NUMBER",
    CREDIT_DEBIT_CARD_CVV: "CREDIT_DEBIT_CARD_CVV",
    CREDIT_DEBIT_CARD_EXPIRY: "CREDIT_DEBIT_CARD_EXPIRY",
    CREDIT_DEBIT_CARD_NUMBER: "CREDIT_DEBIT_CARD_NUMBER",
    DRIVER_ID: "DRIVER_ID",
    EMAIL: "EMAIL",
    INTERNATIONAL_BANK_ACCOUNT_NUMBER: "INTERNATIONAL_BANK_ACCOUNT_NUMBER",
    IP_ADDRESS: "IP_ADDRESS",
    LICENSE_PLATE: "LICENSE_PLATE",
    MAC_ADDRESS: "MAC_ADDRESS",
    NAME: "NAME",
    PASSWORD: "PASSWORD",
    PHONE: "PHONE",
    PIN: "PIN",
    SWIFT_CODE: "SWIFT_CODE",
    UK_NATIONAL_HEALTH_SERVICE_NUMBER: "UK_NATIONAL_HEALTH_SERVICE_NUMBER",
    UK_NATIONAL_INSURANCE_NUMBER: "UK_NATIONAL_INSURANCE_NUMBER",
    UK_UNIQUE_TAXPAYER_REFERENCE_NUMBER: "UK_UNIQUE_TAXPAYER_REFERENCE_NUMBER",
    URL: "URL",
    USERNAME: "USERNAME",
    US_BANK_ACCOUNT_NUMBER: "US_BANK_ACCOUNT_NUMBER",
    US_BANK_ROUTING_NUMBER: "US_BANK_ROUTING_NUMBER",
    US_INDIVIDUAL_TAX_IDENTIFICATION_NUMBER: "US_INDIVIDUAL_TAX_IDENTIFICATION_NUMBER",
    US_PASSPORT_NUMBER: "US_PASSPORT_NUMBER",
    US_SOCIAL_SECURITY_NUMBER: "US_SOCIAL_SECURITY_NUMBER",
    VEHICLE_IDENTIFICATION_NUMBER: "VEHICLE_IDENTIFICATION_NUMBER",
};
const GuardrailTopicPolicyAction = {
    BLOCKED: "BLOCKED",
};
const GuardrailTopicType = {
    DENY: "DENY",
};
const GuardrailWordPolicyAction = {
    BLOCKED: "BLOCKED",
};
const GuardrailManagedWordType = {
    PROFANITY: "PROFANITY",
};
const InvocationType = {
    ACTION_GROUP: "ACTION_GROUP",
    ACTION_GROUP_CODE_INTERPRETER: "ACTION_GROUP_CODE_INTERPRETER",
    AGENT_COLLABORATOR: "AGENT_COLLABORATOR",
    FINISH: "FINISH",
    KNOWLEDGE_BASE: "KNOWLEDGE_BASE",
};
const CreationMode = {
    DEFAULT: "DEFAULT",
    OVERRIDDEN: "OVERRIDDEN",
};
const PromptType = {
    KNOWLEDGE_BASE_RESPONSE_GENERATION: "KNOWLEDGE_BASE_RESPONSE_GENERATION",
    ORCHESTRATION: "ORCHESTRATION",
    POST_PROCESSING: "POST_PROCESSING",
    PRE_PROCESSING: "PRE_PROCESSING",
    ROUTING_CLASSIFIER: "ROUTING_CLASSIFIER",
};
const RetrievalResultContentColumnType = {
    BLOB: "BLOB",
    BOOLEAN: "BOOLEAN",
    DOUBLE: "DOUBLE",
    LONG: "LONG",
    NULL: "NULL",
    STRING: "STRING",
};
const RetrievalResultContentType = {
    AUDIO: "AUDIO",
    IMAGE: "IMAGE",
    ROW: "ROW",
    TEXT: "TEXT",
    VIDEO: "VIDEO",
};
const RetrievalResultLocationType = {
    CONFLUENCE: "CONFLUENCE",
    CUSTOM: "CUSTOM",
    KENDRA: "KENDRA",
    S3: "S3",
    SALESFORCE: "SALESFORCE",
    SHAREPOINT: "SHAREPOINT",
    SQL: "SQL",
    WEB: "WEB",
};
const Source = {
    ACTION_GROUP: "ACTION_GROUP",
    KNOWLEDGE_BASE: "KNOWLEDGE_BASE",
    PARSER: "PARSER",
};
const Type = {
    ACTION_GROUP: "ACTION_GROUP",
    AGENT_COLLABORATOR: "AGENT_COLLABORATOR",
    ASK_USER: "ASK_USER",
    FINISH: "FINISH",
    KNOWLEDGE_BASE: "KNOWLEDGE_BASE",
    REPROMPT: "REPROMPT",
};
const FlowExecutionErrorType = {
    TIMED_OUT: "ExecutionTimedOut",
};
const FlowExecutionStatus = {
    ABORTED: "Aborted",
    FAILED: "Failed",
    RUNNING: "Running",
    SUCCEEDED: "Succeeded",
    TIMED_OUT: "TimedOut",
};
const FlowExecutionEventType = {
    FLOW: "Flow",
    NODE: "Node",
};
const FlowErrorCode = {
    INTERNAL_SERVER: "INTERNAL_SERVER",
    NODE_EXECUTION_FAILED: "NODE_EXECUTION_FAILED",
    VALIDATION: "VALIDATION",
};
const NodeErrorCode = {
    BAD_GATEWAY: "BAD_GATEWAY",
    DEPENDENCY_FAILED: "DEPENDENCY_FAILED",
    INTERNAL_SERVER: "INTERNAL_SERVER",
    VALIDATION: "VALIDATION",
};
const FlowNodeInputCategory = {
    EXIT_LOOP: "ExitLoop",
    LOOP_CONDITION: "LoopCondition",
    RETURN_VALUE_TO_LOOP_START: "ReturnValueToLoopStart",
};
const FlowControlNodeType = {
    ITERATOR: "Iterator",
    LOOP: "Loop",
};
const FlowNodeIODataType = {
    ARRAY: "Array",
    BOOLEAN: "Boolean",
    NUMBER: "Number",
    OBJECT: "Object",
    STRING: "String",
};
const PerformanceConfigLatency = {
    OPTIMIZED: "optimized",
    STANDARD: "standard",
};
const FlowCompletionReason = {
    INPUT_REQUIRED: "INPUT_REQUIRED",
    SUCCESS: "SUCCESS",
};
const NodeType = {
    CONDITION_NODE: "ConditionNode",
    FLOW_INPUT_NODE: "FlowInputNode",
    FLOW_OUTPUT_NODE: "FlowOutputNode",
    KNOWLEDGE_BASE_NODE: "KnowledgeBaseNode",
    LAMBDA_FUNCTION_NODE: "LambdaFunctionNode",
    LEX_NODE: "LexNode",
    PROMPT_NODE: "PromptNode",
};
const InputQueryType = {
    TEXT: "TEXT",
};
const QueryTransformationMode = {
    TEXT_TO_SQL: "TEXT_TO_SQL",
};
const TextToSqlConfigurationType = {
    KNOWLEDGE_BASE: "KNOWLEDGE_BASE",
};
const GeneratedQueryType = {
    REDSHIFT_SQL: "REDSHIFT_SQL",
};
const ConversationRole = {
    ASSISTANT: "assistant",
    USER: "user",
};
const FileSourceType = {
    BYTE_CONTENT: "BYTE_CONTENT",
    S3: "S3",
};
const FileUseCase = {
    CHAT: "CHAT",
    CODE_INTERPRETER: "CODE_INTERPRETER",
};
const AttributeType = {
    BOOLEAN: "BOOLEAN",
    NUMBER: "NUMBER",
    STRING: "STRING",
    STRING_LIST: "STRING_LIST",
};
const SearchType = {
    HYBRID: "HYBRID",
    SEMANTIC: "SEMANTIC",
};
const RerankingMetadataSelectionMode = {
    ALL: "ALL",
    SELECTIVE: "SELECTIVE",
};
const VectorSearchRerankingConfigurationType = {
    BEDROCK_RERANKING_MODEL: "BEDROCK_RERANKING_MODEL",
};
const RelayConversationHistory = {
    DISABLED: "DISABLED",
    TO_COLLABORATOR: "TO_COLLABORATOR",
};
const PromptState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const OrchestrationType = {
    CUSTOM_ORCHESTRATION: "CUSTOM_ORCHESTRATION",
    DEFAULT: "DEFAULT",
};
const MemoryType = {
    SESSION_SUMMARY: "SESSION_SUMMARY",
};
const RerankQueryContentType = {
    TEXT: "TEXT",
};
const RerankingConfigurationType = {
    BEDROCK_RERANKING_MODEL: "BEDROCK_RERANKING_MODEL",
};
const RerankDocumentType = {
    JSON: "JSON",
    TEXT: "TEXT",
};
const RerankSourceType = {
    INLINE: "INLINE",
};
const ExternalSourceType = {
    BYTE_CONTENT: "BYTE_CONTENT",
    S3: "S3",
};
const QueryTransformationType = {
    QUERY_DECOMPOSITION: "QUERY_DECOMPOSITION",
};
const RetrieveAndGenerateType = {
    EXTERNAL_SOURCES: "EXTERNAL_SOURCES",
    KNOWLEDGE_BASE: "KNOWLEDGE_BASE",
};
const GuadrailAction = {
    INTERVENED: "INTERVENED",
    NONE: "NONE",
};
const InputImageFormat = {
    gif: "gif",
    jpeg: "jpeg",
    png: "png",
    webp: "webp",
};
const KnowledgeBaseQueryType = {
    IMAGE: "IMAGE",
    TEXT: "TEXT",
};
const SessionStatus = {
    ACTIVE: "ACTIVE",
    ENDED: "ENDED",
    EXPIRED: "EXPIRED",
};
const ImageFormat = {
    GIF: "gif",
    JPEG: "jpeg",
    PNG: "png",
    WEBP: "webp",
};

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
exports.APISchema$ = APISchema$;
exports.AccessDeniedException = AccessDeniedException;
exports.AccessDeniedException$ = AccessDeniedException$;
exports.ActionGroupExecutor$ = ActionGroupExecutor$;
exports.ActionGroupInvocationInput$ = ActionGroupInvocationInput$;
exports.ActionGroupInvocationOutput$ = ActionGroupInvocationOutput$;
exports.ActionGroupSignature = ActionGroupSignature;
exports.ActionInvocationType = ActionInvocationType;
exports.AgentActionGroup$ = AgentActionGroup$;
exports.AgentCollaboration = AgentCollaboration;
exports.AgentCollaboratorInputPayload$ = AgentCollaboratorInputPayload$;
exports.AgentCollaboratorInvocationInput$ = AgentCollaboratorInvocationInput$;
exports.AgentCollaboratorInvocationOutput$ = AgentCollaboratorInvocationOutput$;
exports.AgentCollaboratorOutputPayload$ = AgentCollaboratorOutputPayload$;
exports.AnalyzePromptEvent$ = AnalyzePromptEvent$;
exports.ApiInvocationInput$ = ApiInvocationInput$;
exports.ApiParameter$ = ApiParameter$;
exports.ApiRequestBody$ = ApiRequestBody$;
exports.ApiResult$ = ApiResult$;
exports.AttributeType = AttributeType;
exports.Attribution$ = Attribution$;
exports.AudioSegment$ = AudioSegment$;
exports.BadGatewayException = BadGatewayException;
exports.BadGatewayException$ = BadGatewayException$;
exports.BedrockAgentRuntime = BedrockAgentRuntime;
exports.BedrockAgentRuntimeClient = BedrockAgentRuntimeClient;
exports.BedrockAgentRuntimeServiceException = BedrockAgentRuntimeServiceException;
exports.BedrockAgentRuntimeServiceException$ = BedrockAgentRuntimeServiceException$;
exports.BedrockModelConfigurations$ = BedrockModelConfigurations$;
exports.BedrockRerankingConfiguration$ = BedrockRerankingConfiguration$;
exports.BedrockRerankingModelConfiguration$ = BedrockRerankingModelConfiguration$;
exports.BedrockSessionContentBlock$ = BedrockSessionContentBlock$;
exports.ByteContentDoc$ = ByteContentDoc$;
exports.ByteContentFile$ = ByteContentFile$;
exports.Caller$ = Caller$;
exports.Citation$ = Citation$;
exports.CitationEvent$ = CitationEvent$;
exports.CodeInterpreterInvocationInput$ = CodeInterpreterInvocationInput$;
exports.CodeInterpreterInvocationOutput$ = CodeInterpreterInvocationOutput$;
exports.Collaborator$ = Collaborator$;
exports.CollaboratorConfiguration$ = CollaboratorConfiguration$;
exports.ConditionResultEvent$ = ConditionResultEvent$;
exports.ConfirmationState = ConfirmationState;
exports.ConflictException = ConflictException;
exports.ConflictException$ = ConflictException$;
exports.ContentBlock$ = ContentBlock$;
exports.ContentBody$ = ContentBody$;
exports.ConversationHistory$ = ConversationHistory$;
exports.ConversationRole = ConversationRole;
exports.CreateInvocation$ = CreateInvocation$;
exports.CreateInvocationCommand = CreateInvocationCommand;
exports.CreateInvocationRequest$ = CreateInvocationRequest$;
exports.CreateInvocationResponse$ = CreateInvocationResponse$;
exports.CreateSession$ = CreateSession$;
exports.CreateSessionCommand = CreateSessionCommand;
exports.CreateSessionRequest$ = CreateSessionRequest$;
exports.CreateSessionResponse$ = CreateSessionResponse$;
exports.CreationMode = CreationMode;
exports.CustomControlMethod = CustomControlMethod;
exports.CustomOrchestration$ = CustomOrchestration$;
exports.CustomOrchestrationTrace$ = CustomOrchestrationTrace$;
exports.CustomOrchestrationTraceEvent$ = CustomOrchestrationTraceEvent$;
exports.DeleteAgentMemory$ = DeleteAgentMemory$;
exports.DeleteAgentMemoryCommand = DeleteAgentMemoryCommand;
exports.DeleteAgentMemoryRequest$ = DeleteAgentMemoryRequest$;
exports.DeleteAgentMemoryResponse$ = DeleteAgentMemoryResponse$;
exports.DeleteSession$ = DeleteSession$;
exports.DeleteSessionCommand = DeleteSessionCommand;
exports.DeleteSessionRequest$ = DeleteSessionRequest$;
exports.DeleteSessionResponse$ = DeleteSessionResponse$;
exports.DependencyFailedException = DependencyFailedException;
exports.DependencyFailedException$ = DependencyFailedException$;
exports.EndSession$ = EndSession$;
exports.EndSessionCommand = EndSessionCommand;
exports.EndSessionRequest$ = EndSessionRequest$;
exports.EndSessionResponse$ = EndSessionResponse$;
exports.ExecutionType = ExecutionType;
exports.ExternalSource$ = ExternalSource$;
exports.ExternalSourceType = ExternalSourceType;
exports.ExternalSourcesGenerationConfiguration$ = ExternalSourcesGenerationConfiguration$;
exports.ExternalSourcesRetrieveAndGenerateConfiguration$ = ExternalSourcesRetrieveAndGenerateConfiguration$;
exports.FailureTrace$ = FailureTrace$;
exports.FieldForReranking$ = FieldForReranking$;
exports.FilePart$ = FilePart$;
exports.FileSource$ = FileSource$;
exports.FileSourceType = FileSourceType;
exports.FileUseCase = FileUseCase;
exports.FilterAttribute$ = FilterAttribute$;
exports.FinalResponse$ = FinalResponse$;
exports.FlowCompletionEvent$ = FlowCompletionEvent$;
exports.FlowCompletionReason = FlowCompletionReason;
exports.FlowControlNodeType = FlowControlNodeType;
exports.FlowErrorCode = FlowErrorCode;
exports.FlowExecutionContent$ = FlowExecutionContent$;
exports.FlowExecutionError$ = FlowExecutionError$;
exports.FlowExecutionErrorType = FlowExecutionErrorType;
exports.FlowExecutionEvent$ = FlowExecutionEvent$;
exports.FlowExecutionEventType = FlowExecutionEventType;
exports.FlowExecutionInputEvent$ = FlowExecutionInputEvent$;
exports.FlowExecutionOutputEvent$ = FlowExecutionOutputEvent$;
exports.FlowExecutionStatus = FlowExecutionStatus;
exports.FlowExecutionSummary$ = FlowExecutionSummary$;
exports.FlowFailureEvent$ = FlowFailureEvent$;
exports.FlowInput$ = FlowInput$;
exports.FlowInputContent$ = FlowInputContent$;
exports.FlowInputField$ = FlowInputField$;
exports.FlowMultiTurnInputContent$ = FlowMultiTurnInputContent$;
exports.FlowMultiTurnInputRequestEvent$ = FlowMultiTurnInputRequestEvent$;
exports.FlowNodeIODataType = FlowNodeIODataType;
exports.FlowNodeInputCategory = FlowNodeInputCategory;
exports.FlowOutputContent$ = FlowOutputContent$;
exports.FlowOutputEvent$ = FlowOutputEvent$;
exports.FlowOutputField$ = FlowOutputField$;
exports.FlowResponseStream$ = FlowResponseStream$;
exports.FlowTrace$ = FlowTrace$;
exports.FlowTraceCondition$ = FlowTraceCondition$;
exports.FlowTraceConditionNodeResultEvent$ = FlowTraceConditionNodeResultEvent$;
exports.FlowTraceDependencyEvent$ = FlowTraceDependencyEvent$;
exports.FlowTraceEvent$ = FlowTraceEvent$;
exports.FlowTraceNodeActionEvent$ = FlowTraceNodeActionEvent$;
exports.FlowTraceNodeInputContent$ = FlowTraceNodeInputContent$;
exports.FlowTraceNodeInputEvent$ = FlowTraceNodeInputEvent$;
exports.FlowTraceNodeInputExecutionChainItem$ = FlowTraceNodeInputExecutionChainItem$;
exports.FlowTraceNodeInputField$ = FlowTraceNodeInputField$;
exports.FlowTraceNodeInputSource$ = FlowTraceNodeInputSource$;
exports.FlowTraceNodeOutputContent$ = FlowTraceNodeOutputContent$;
exports.FlowTraceNodeOutputEvent$ = FlowTraceNodeOutputEvent$;
exports.FlowTraceNodeOutputField$ = FlowTraceNodeOutputField$;
exports.FlowTraceNodeOutputNext$ = FlowTraceNodeOutputNext$;
exports.FunctionDefinition$ = FunctionDefinition$;
exports.FunctionInvocationInput$ = FunctionInvocationInput$;
exports.FunctionParameter$ = FunctionParameter$;
exports.FunctionResult$ = FunctionResult$;
exports.FunctionSchema$ = FunctionSchema$;
exports.GenerateQuery$ = GenerateQuery$;
exports.GenerateQueryCommand = GenerateQueryCommand;
exports.GenerateQueryRequest$ = GenerateQueryRequest$;
exports.GenerateQueryResponse$ = GenerateQueryResponse$;
exports.GeneratedQuery$ = GeneratedQuery$;
exports.GeneratedQueryType = GeneratedQueryType;
exports.GeneratedResponsePart$ = GeneratedResponsePart$;
exports.GenerationConfiguration$ = GenerationConfiguration$;
exports.GetAgentMemory$ = GetAgentMemory$;
exports.GetAgentMemoryCommand = GetAgentMemoryCommand;
exports.GetAgentMemoryRequest$ = GetAgentMemoryRequest$;
exports.GetAgentMemoryResponse$ = GetAgentMemoryResponse$;
exports.GetExecutionFlowSnapshot$ = GetExecutionFlowSnapshot$;
exports.GetExecutionFlowSnapshotCommand = GetExecutionFlowSnapshotCommand;
exports.GetExecutionFlowSnapshotRequest$ = GetExecutionFlowSnapshotRequest$;
exports.GetExecutionFlowSnapshotResponse$ = GetExecutionFlowSnapshotResponse$;
exports.GetFlowExecution$ = GetFlowExecution$;
exports.GetFlowExecutionCommand = GetFlowExecutionCommand;
exports.GetFlowExecutionRequest$ = GetFlowExecutionRequest$;
exports.GetFlowExecutionResponse$ = GetFlowExecutionResponse$;
exports.GetInvocationStep$ = GetInvocationStep$;
exports.GetInvocationStepCommand = GetInvocationStepCommand;
exports.GetInvocationStepRequest$ = GetInvocationStepRequest$;
exports.GetInvocationStepResponse$ = GetInvocationStepResponse$;
exports.GetSession$ = GetSession$;
exports.GetSessionCommand = GetSessionCommand;
exports.GetSessionRequest$ = GetSessionRequest$;
exports.GetSessionResponse$ = GetSessionResponse$;
exports.GuadrailAction = GuadrailAction;
exports.GuardrailAction = GuardrailAction;
exports.GuardrailAssessment$ = GuardrailAssessment$;
exports.GuardrailConfiguration$ = GuardrailConfiguration$;
exports.GuardrailConfigurationWithArn$ = GuardrailConfigurationWithArn$;
exports.GuardrailContentFilter$ = GuardrailContentFilter$;
exports.GuardrailContentFilterConfidence = GuardrailContentFilterConfidence;
exports.GuardrailContentFilterType = GuardrailContentFilterType;
exports.GuardrailContentPolicyAction = GuardrailContentPolicyAction;
exports.GuardrailContentPolicyAssessment$ = GuardrailContentPolicyAssessment$;
exports.GuardrailCustomWord$ = GuardrailCustomWord$;
exports.GuardrailEvent$ = GuardrailEvent$;
exports.GuardrailManagedWord$ = GuardrailManagedWord$;
exports.GuardrailManagedWordType = GuardrailManagedWordType;
exports.GuardrailPiiEntityFilter$ = GuardrailPiiEntityFilter$;
exports.GuardrailPiiEntityType = GuardrailPiiEntityType;
exports.GuardrailRegexFilter$ = GuardrailRegexFilter$;
exports.GuardrailSensitiveInformationPolicyAction = GuardrailSensitiveInformationPolicyAction;
exports.GuardrailSensitiveInformationPolicyAssessment$ = GuardrailSensitiveInformationPolicyAssessment$;
exports.GuardrailTopic$ = GuardrailTopic$;
exports.GuardrailTopicPolicyAction = GuardrailTopicPolicyAction;
exports.GuardrailTopicPolicyAssessment$ = GuardrailTopicPolicyAssessment$;
exports.GuardrailTopicType = GuardrailTopicType;
exports.GuardrailTrace$ = GuardrailTrace$;
exports.GuardrailWordPolicyAction = GuardrailWordPolicyAction;
exports.GuardrailWordPolicyAssessment$ = GuardrailWordPolicyAssessment$;
exports.ImageBlock$ = ImageBlock$;
exports.ImageFormat = ImageFormat;
exports.ImageInput$ = ImageInput$;
exports.ImageInputFormat = ImageInputFormat;
exports.ImageInputSource$ = ImageInputSource$;
exports.ImageSource$ = ImageSource$;
exports.ImplicitFilterConfiguration$ = ImplicitFilterConfiguration$;
exports.InferenceConfig$ = InferenceConfig$;
exports.InferenceConfiguration$ = InferenceConfiguration$;
exports.InlineAgentFilePart$ = InlineAgentFilePart$;
exports.InlineAgentPayloadPart$ = InlineAgentPayloadPart$;
exports.InlineAgentResponseStream$ = InlineAgentResponseStream$;
exports.InlineAgentReturnControlPayload$ = InlineAgentReturnControlPayload$;
exports.InlineAgentTracePart$ = InlineAgentTracePart$;
exports.InlineBedrockModelConfigurations$ = InlineBedrockModelConfigurations$;
exports.InlineSessionState$ = InlineSessionState$;
exports.InputFile$ = InputFile$;
exports.InputImage$ = InputImage$;
exports.InputImageFormat = InputImageFormat;
exports.InputPrompt$ = InputPrompt$;
exports.InputQueryType = InputQueryType;
exports.InternalServerException = InternalServerException;
exports.InternalServerException$ = InternalServerException$;
exports.InvocationInput$ = InvocationInput$;
exports.InvocationInputMember$ = InvocationInputMember$;
exports.InvocationResultMember$ = InvocationResultMember$;
exports.InvocationStep$ = InvocationStep$;
exports.InvocationStepPayload$ = InvocationStepPayload$;
exports.InvocationStepSummary$ = InvocationStepSummary$;
exports.InvocationSummary$ = InvocationSummary$;
exports.InvocationType = InvocationType;
exports.InvokeAgent$ = InvokeAgent$;
exports.InvokeAgentCommand = InvokeAgentCommand;
exports.InvokeAgentRequest$ = InvokeAgentRequest$;
exports.InvokeAgentResponse$ = InvokeAgentResponse$;
exports.InvokeFlow$ = InvokeFlow$;
exports.InvokeFlowCommand = InvokeFlowCommand;
exports.InvokeFlowRequest$ = InvokeFlowRequest$;
exports.InvokeFlowResponse$ = InvokeFlowResponse$;
exports.InvokeInlineAgent$ = InvokeInlineAgent$;
exports.InvokeInlineAgentCommand = InvokeInlineAgentCommand;
exports.InvokeInlineAgentRequest$ = InvokeInlineAgentRequest$;
exports.InvokeInlineAgentResponse$ = InvokeInlineAgentResponse$;
exports.KnowledgeBase$ = KnowledgeBase$;
exports.KnowledgeBaseConfiguration$ = KnowledgeBaseConfiguration$;
exports.KnowledgeBaseLookupInput$ = KnowledgeBaseLookupInput$;
exports.KnowledgeBaseLookupOutput$ = KnowledgeBaseLookupOutput$;
exports.KnowledgeBaseQuery$ = KnowledgeBaseQuery$;
exports.KnowledgeBaseQueryType = KnowledgeBaseQueryType;
exports.KnowledgeBaseRetrievalConfiguration$ = KnowledgeBaseRetrievalConfiguration$;
exports.KnowledgeBaseRetrievalResult$ = KnowledgeBaseRetrievalResult$;
exports.KnowledgeBaseRetrieveAndGenerateConfiguration$ = KnowledgeBaseRetrieveAndGenerateConfiguration$;
exports.KnowledgeBaseVectorSearchConfiguration$ = KnowledgeBaseVectorSearchConfiguration$;
exports.ListFlowExecutionEvents$ = ListFlowExecutionEvents$;
exports.ListFlowExecutionEventsCommand = ListFlowExecutionEventsCommand;
exports.ListFlowExecutionEventsRequest$ = ListFlowExecutionEventsRequest$;
exports.ListFlowExecutionEventsResponse$ = ListFlowExecutionEventsResponse$;
exports.ListFlowExecutions$ = ListFlowExecutions$;
exports.ListFlowExecutionsCommand = ListFlowExecutionsCommand;
exports.ListFlowExecutionsRequest$ = ListFlowExecutionsRequest$;
exports.ListFlowExecutionsResponse$ = ListFlowExecutionsResponse$;
exports.ListInvocationSteps$ = ListInvocationSteps$;
exports.ListInvocationStepsCommand = ListInvocationStepsCommand;
exports.ListInvocationStepsRequest$ = ListInvocationStepsRequest$;
exports.ListInvocationStepsResponse$ = ListInvocationStepsResponse$;
exports.ListInvocations$ = ListInvocations$;
exports.ListInvocationsCommand = ListInvocationsCommand;
exports.ListInvocationsRequest$ = ListInvocationsRequest$;
exports.ListInvocationsResponse$ = ListInvocationsResponse$;
exports.ListSessions$ = ListSessions$;
exports.ListSessionsCommand = ListSessionsCommand;
exports.ListSessionsRequest$ = ListSessionsRequest$;
exports.ListSessionsResponse$ = ListSessionsResponse$;
exports.ListTagsForResource$ = ListTagsForResource$;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListTagsForResourceRequest$ = ListTagsForResourceRequest$;
exports.ListTagsForResourceResponse$ = ListTagsForResourceResponse$;
exports.Memory$ = Memory$;
exports.MemorySessionSummary$ = MemorySessionSummary$;
exports.MemoryType = MemoryType;
exports.Message$ = Message$;
exports.Metadata$ = Metadata$;
exports.MetadataAttributeSchema$ = MetadataAttributeSchema$;
exports.MetadataConfigurationForReranking$ = MetadataConfigurationForReranking$;
exports.ModelInvocationInput$ = ModelInvocationInput$;
exports.ModelNotReadyException = ModelNotReadyException;
exports.ModelNotReadyException$ = ModelNotReadyException$;
exports.ModelPerformanceConfiguration$ = ModelPerformanceConfiguration$;
exports.NodeActionEvent$ = NodeActionEvent$;
exports.NodeDependencyEvent$ = NodeDependencyEvent$;
exports.NodeErrorCode = NodeErrorCode;
exports.NodeExecutionContent$ = NodeExecutionContent$;
exports.NodeFailureEvent$ = NodeFailureEvent$;
exports.NodeInputEvent$ = NodeInputEvent$;
exports.NodeInputExecutionChainItem$ = NodeInputExecutionChainItem$;
exports.NodeInputField$ = NodeInputField$;
exports.NodeInputSource$ = NodeInputSource$;
exports.NodeOutputEvent$ = NodeOutputEvent$;
exports.NodeOutputField$ = NodeOutputField$;
exports.NodeOutputNext$ = NodeOutputNext$;
exports.NodeTraceElements$ = NodeTraceElements$;
exports.NodeType = NodeType;
exports.Observation$ = Observation$;
exports.OptimizePrompt$ = OptimizePrompt$;
exports.OptimizePromptCommand = OptimizePromptCommand;
exports.OptimizePromptRequest$ = OptimizePromptRequest$;
exports.OptimizePromptResponse$ = OptimizePromptResponse$;
exports.OptimizedPrompt$ = OptimizedPrompt$;
exports.OptimizedPromptEvent$ = OptimizedPromptEvent$;
exports.OptimizedPromptStream$ = OptimizedPromptStream$;
exports.OrchestrationConfiguration$ = OrchestrationConfiguration$;
exports.OrchestrationExecutor$ = OrchestrationExecutor$;
exports.OrchestrationModelInvocationOutput$ = OrchestrationModelInvocationOutput$;
exports.OrchestrationTrace$ = OrchestrationTrace$;
exports.OrchestrationType = OrchestrationType;
exports.OutputFile$ = OutputFile$;
exports.Parameter$ = Parameter$;
exports.ParameterDetail$ = ParameterDetail$;
exports.ParameterType = ParameterType;
exports.PayloadPart$ = PayloadPart$;
exports.PayloadType = PayloadType;
exports.PerformanceConfigLatency = PerformanceConfigLatency;
exports.PerformanceConfiguration$ = PerformanceConfiguration$;
exports.PostProcessingModelInvocationOutput$ = PostProcessingModelInvocationOutput$;
exports.PostProcessingParsedResponse$ = PostProcessingParsedResponse$;
exports.PostProcessingTrace$ = PostProcessingTrace$;
exports.PreProcessingModelInvocationOutput$ = PreProcessingModelInvocationOutput$;
exports.PreProcessingParsedResponse$ = PreProcessingParsedResponse$;
exports.PreProcessingTrace$ = PreProcessingTrace$;
exports.PromptConfiguration$ = PromptConfiguration$;
exports.PromptCreationConfigurations$ = PromptCreationConfigurations$;
exports.PromptOverrideConfiguration$ = PromptOverrideConfiguration$;
exports.PromptState = PromptState;
exports.PromptTemplate$ = PromptTemplate$;
exports.PromptType = PromptType;
exports.PropertyParameters$ = PropertyParameters$;
exports.PutInvocationStep$ = PutInvocationStep$;
exports.PutInvocationStepCommand = PutInvocationStepCommand;
exports.PutInvocationStepRequest$ = PutInvocationStepRequest$;
exports.PutInvocationStepResponse$ = PutInvocationStepResponse$;
exports.QueryGenerationInput$ = QueryGenerationInput$;
exports.QueryTransformationConfiguration$ = QueryTransformationConfiguration$;
exports.QueryTransformationMode = QueryTransformationMode;
exports.QueryTransformationType = QueryTransformationType;
exports.Rationale$ = Rationale$;
exports.RawResponse$ = RawResponse$;
exports.ReasoningContentBlock$ = ReasoningContentBlock$;
exports.ReasoningTextBlock$ = ReasoningTextBlock$;
exports.RelayConversationHistory = RelayConversationHistory;
exports.RepromptResponse$ = RepromptResponse$;
exports.RequestBody$ = RequestBody$;
exports.RequireConfirmation = RequireConfirmation;
exports.Rerank$ = Rerank$;
exports.RerankCommand = RerankCommand;
exports.RerankDocument$ = RerankDocument$;
exports.RerankDocumentType = RerankDocumentType;
exports.RerankQuery$ = RerankQuery$;
exports.RerankQueryContentType = RerankQueryContentType;
exports.RerankRequest$ = RerankRequest$;
exports.RerankResponse$ = RerankResponse$;
exports.RerankResult$ = RerankResult$;
exports.RerankSource$ = RerankSource$;
exports.RerankSourceType = RerankSourceType;
exports.RerankTextDocument$ = RerankTextDocument$;
exports.RerankingConfiguration$ = RerankingConfiguration$;
exports.RerankingConfigurationType = RerankingConfigurationType;
exports.RerankingMetadataSelectionMode = RerankingMetadataSelectionMode;
exports.RerankingMetadataSelectiveModeConfiguration$ = RerankingMetadataSelectiveModeConfiguration$;
exports.ResourceNotFoundException = ResourceNotFoundException;
exports.ResourceNotFoundException$ = ResourceNotFoundException$;
exports.ResponseState = ResponseState;
exports.ResponseStream$ = ResponseStream$;
exports.RetrievalFilter$ = RetrievalFilter$;
exports.RetrievalResultConfluenceLocation$ = RetrievalResultConfluenceLocation$;
exports.RetrievalResultContent$ = RetrievalResultContent$;
exports.RetrievalResultContentColumn$ = RetrievalResultContentColumn$;
exports.RetrievalResultContentColumnType = RetrievalResultContentColumnType;
exports.RetrievalResultContentType = RetrievalResultContentType;
exports.RetrievalResultCustomDocumentLocation$ = RetrievalResultCustomDocumentLocation$;
exports.RetrievalResultKendraDocumentLocation$ = RetrievalResultKendraDocumentLocation$;
exports.RetrievalResultLocation$ = RetrievalResultLocation$;
exports.RetrievalResultLocationType = RetrievalResultLocationType;
exports.RetrievalResultS3Location$ = RetrievalResultS3Location$;
exports.RetrievalResultSalesforceLocation$ = RetrievalResultSalesforceLocation$;
exports.RetrievalResultSharePointLocation$ = RetrievalResultSharePointLocation$;
exports.RetrievalResultSqlLocation$ = RetrievalResultSqlLocation$;
exports.RetrievalResultWebLocation$ = RetrievalResultWebLocation$;
exports.Retrieve$ = Retrieve$;
exports.RetrieveAndGenerate$ = RetrieveAndGenerate$;
exports.RetrieveAndGenerateCommand = RetrieveAndGenerateCommand;
exports.RetrieveAndGenerateConfiguration$ = RetrieveAndGenerateConfiguration$;
exports.RetrieveAndGenerateInput$ = RetrieveAndGenerateInput$;
exports.RetrieveAndGenerateOutput$ = RetrieveAndGenerateOutput$;
exports.RetrieveAndGenerateOutputEvent$ = RetrieveAndGenerateOutputEvent$;
exports.RetrieveAndGenerateRequest$ = RetrieveAndGenerateRequest$;
exports.RetrieveAndGenerateResponse$ = RetrieveAndGenerateResponse$;
exports.RetrieveAndGenerateSessionConfiguration$ = RetrieveAndGenerateSessionConfiguration$;
exports.RetrieveAndGenerateStream$ = RetrieveAndGenerateStream$;
exports.RetrieveAndGenerateStreamCommand = RetrieveAndGenerateStreamCommand;
exports.RetrieveAndGenerateStreamRequest$ = RetrieveAndGenerateStreamRequest$;
exports.RetrieveAndGenerateStreamResponse$ = RetrieveAndGenerateStreamResponse$;
exports.RetrieveAndGenerateStreamResponseOutput$ = RetrieveAndGenerateStreamResponseOutput$;
exports.RetrieveAndGenerateType = RetrieveAndGenerateType;
exports.RetrieveCommand = RetrieveCommand;
exports.RetrieveRequest$ = RetrieveRequest$;
exports.RetrieveResponse$ = RetrieveResponse$;
exports.RetrievedReference$ = RetrievedReference$;
exports.ReturnControlPayload$ = ReturnControlPayload$;
exports.ReturnControlResults$ = ReturnControlResults$;
exports.RoutingClassifierModelInvocationOutput$ = RoutingClassifierModelInvocationOutput$;
exports.RoutingClassifierTrace$ = RoutingClassifierTrace$;
exports.S3Identifier$ = S3Identifier$;
exports.S3Location$ = S3Location$;
exports.S3ObjectDoc$ = S3ObjectDoc$;
exports.S3ObjectFile$ = S3ObjectFile$;
exports.SatisfiedCondition$ = SatisfiedCondition$;
exports.SearchType = SearchType;
exports.ServiceQuotaExceededException = ServiceQuotaExceededException;
exports.ServiceQuotaExceededException$ = ServiceQuotaExceededException$;
exports.SessionState$ = SessionState$;
exports.SessionStatus = SessionStatus;
exports.SessionSummary$ = SessionSummary$;
exports.Source = Source;
exports.Span$ = Span$;
exports.StartFlowExecution$ = StartFlowExecution$;
exports.StartFlowExecutionCommand = StartFlowExecutionCommand;
exports.StartFlowExecutionRequest$ = StartFlowExecutionRequest$;
exports.StartFlowExecutionResponse$ = StartFlowExecutionResponse$;
exports.StopFlowExecution$ = StopFlowExecution$;
exports.StopFlowExecutionCommand = StopFlowExecutionCommand;
exports.StopFlowExecutionRequest$ = StopFlowExecutionRequest$;
exports.StopFlowExecutionResponse$ = StopFlowExecutionResponse$;
exports.StreamingConfigurations$ = StreamingConfigurations$;
exports.TagResource$ = TagResource$;
exports.TagResourceCommand = TagResourceCommand;
exports.TagResourceRequest$ = TagResourceRequest$;
exports.TagResourceResponse$ = TagResourceResponse$;
exports.TextInferenceConfig$ = TextInferenceConfig$;
exports.TextPrompt$ = TextPrompt$;
exports.TextResponsePart$ = TextResponsePart$;
exports.TextToSqlConfiguration$ = TextToSqlConfiguration$;
exports.TextToSqlConfigurationType = TextToSqlConfigurationType;
exports.TextToSqlKnowledgeBaseConfiguration$ = TextToSqlKnowledgeBaseConfiguration$;
exports.ThrottlingException = ThrottlingException;
exports.ThrottlingException$ = ThrottlingException$;
exports.Trace$ = Trace$;
exports.TraceElements$ = TraceElements$;
exports.TracePart$ = TracePart$;
exports.TransformationConfiguration$ = TransformationConfiguration$;
exports.Type = Type;
exports.UntagResource$ = UntagResource$;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UntagResourceRequest$ = UntagResourceRequest$;
exports.UntagResourceResponse$ = UntagResourceResponse$;
exports.UpdateSession$ = UpdateSession$;
exports.UpdateSessionCommand = UpdateSessionCommand;
exports.UpdateSessionRequest$ = UpdateSessionRequest$;
exports.UpdateSessionResponse$ = UpdateSessionResponse$;
exports.Usage$ = Usage$;
exports.ValidationException = ValidationException;
exports.ValidationException$ = ValidationException$;
exports.VectorSearchBedrockRerankingConfiguration$ = VectorSearchBedrockRerankingConfiguration$;
exports.VectorSearchBedrockRerankingModelConfiguration$ = VectorSearchBedrockRerankingModelConfiguration$;
exports.VectorSearchRerankingConfiguration$ = VectorSearchRerankingConfiguration$;
exports.VectorSearchRerankingConfigurationType = VectorSearchRerankingConfigurationType;
exports.VideoSegment$ = VideoSegment$;
exports.paginateGetAgentMemory = paginateGetAgentMemory;
exports.paginateListFlowExecutionEvents = paginateListFlowExecutionEvents;
exports.paginateListFlowExecutions = paginateListFlowExecutions;
exports.paginateListInvocationSteps = paginateListInvocationSteps;
exports.paginateListInvocations = paginateListInvocations;
exports.paginateListSessions = paginateListSessions;
exports.paginateRerank = paginateRerank;
exports.paginateRetrieve = paginateRetrieve;
